diff --git a/src/meson.build b/src/meson.build
index 585fe3a..23d57d7 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -410,20 +410,18 @@ endif
 libadwaita = library(
   'adwaita-' + apiversion,
   libadwaita_sources,
-              soversion: soversion,
                  c_args: libadwaita_c_args,
            dependencies: libadwaita_deps,
     include_directories: [ root_inc, src_inc ],
                 install: true,
               link_args: libadwaita_link_args,
             install_dir: adwaita_libdir,
-  gnu_symbol_visibility: 'hidden',
 )
 
 libadwaita_dep = declare_dependency(
               sources: libadwaita_generated_headers,
          dependencies: libadwaita_public_deps,
-            link_with: libadwaita,
+            link_whole: libadwaita,
   include_directories: include_directories('.'),
 )
 
