diff --git a/src/libcamera/source_paths.cpp b/src/libcamera/source_paths.cpp
index 1af5386..352bad6 100644
--- a/src/libcamera/source_paths.cpp
+++ b/src/libcamera/source_paths.cpp
@@ -22,7 +22,7 @@
  */
 
 /* musl doesn't declare _DYNAMIC in link.h, declare it manually. */
-extern ElfW(Dyn) _DYNAMIC[];
+// extern ElfW(Dyn) _DYNAMIC[];
 
 namespace libcamera {
 
@@ -43,10 +43,10 @@ bool isLibcameraInstalled()
 	 * DT_RUNPATH (DT_RPATH when the linker uses old dtags) is removed on
 	 * install.
 	 */
-	for (const ElfW(Dyn) *dyn = _DYNAMIC; dyn->d_tag != DT_NULL; ++dyn) {
-		if (dyn->d_tag == DT_RUNPATH || dyn->d_tag == DT_RPATH)
-			return false;
-	}
+	// for (const ElfW(Dyn) *dyn = _DYNAMIC; dyn->d_tag != DT_NULL; ++dyn) {
+	// 	if (dyn->d_tag == DT_RUNPATH || dyn->d_tag == DT_RPATH)
+	// 		return false;
+	// }
 
 	return true;
 }
