diff --git a/GNUmakefile b/GNUmakefile
index e1403be..47187b3 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -479,8 +479,7 @@ dist_libstemmer_python: $(PYTHON_SOURCES) $(COMMON_FILES)
 	cp -a $(PYTHON_SAMPLE_SOURCES) $${dest}/src/$(python_sample_dir) && \
 	cp -a $(PYTHON_RUNTIME_SOURCES) $${dest}/src/$(python_runtime_dir) && \
 	cp -a $(COMMON_FILES) $(PYTHON_PACKAGE_FILES) $${dest} && \
-	(cd $${dest} && $(python) -m build && cp dist/*.tar.gz dist/*.whl ..) && \
-	rm -rf $${dest}
+	(cd $${dest} && gpep517 build-wheel --wheel-dir dist --output-fd 3 3>&1 >&2)
 
 dist_libstemmer_js: $(JS_SOURCES) $(COMMON_FILES)
 	destname=jsstemmer-$(SNOWBALL_VERSION); \
