diff --git a/src/vulkan/wsi/wsi_common.c b/src/vulkan/wsi/wsi_common.c
index 372c02bc4d3..a1b2bb34952 100644
--- a/src/vulkan/wsi/wsi_common.c
+++ b/src/vulkan/wsi/wsi_common.c
@@ -106,7 +106,9 @@ wsi_device_init(struct wsi_device *wsi,
       .sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_PROPERTIES_2,
       .pNext = &wsi->pci_bus_info,
    };
-   GetPhysicalDeviceProperties2(pdevice, &pdp2);
+   if (GetPhysicalDeviceProperties2){
+      GetPhysicalDeviceProperties2(pdevice, &pdp2);
+   }
 
    wsi->maxImageDimension2D = pdp2.properties.limits.maxImageDimension2D;
    assert(pdp2.properties.limits.optimalBufferCopyRowPitchAlignment <= UINT32_MAX);
