diff --git a/meson.build b/meson.build
index 1bd2bbc..e41821e 100644
--- a/meson.build
+++ b/meson.build
@@ -1522,19 +1522,19 @@ if cc.links('int main() { return 0; }', args : '-Wl,-Bsymbolic', name : 'Bsymbol
   ld_args_bsymbolic += '-Wl,-Bsymbolic'
 endif
 with_ld_version_script = false
-if cc.links('int main() { return 0; }',
-            args : '-Wl,--version-script=@0@'.format(
-              join_paths(meson.current_source_dir(), 'build-support/conftest.map')),
-            name : 'version-script')
-  with_ld_version_script = true
-endif
+# if cc.links('int main() { return 0; }',
+#             args : '-Wl,--version-script=@0@'.format(
+#               join_paths(meson.current_source_dir(), 'build-support/conftest.map')),
+#             name : 'version-script')
+#   with_ld_version_script = true
+# endif
 with_ld_dynamic_list = false
-if cc.links('int main() { return 0; }',
-            args : '-Wl,--dynamic-list=@0@'.format(
-              join_paths(meson.current_source_dir(), 'build-support/conftest.dyn')),
-            name : 'dynamic-list')
-  with_ld_dynamic_list = true
-endif
+# if cc.links('int main() { return 0; }',
+#             args : '-Wl,--dynamic-list=@0@'.format(
+#               join_paths(meson.current_source_dir(), 'build-support/conftest.dyn')),
+#             name : 'dynamic-list')
+#   with_ld_dynamic_list = true
+# endif
 
 ld_args_build_id = cc.get_supported_link_arguments('-Wl,--build-id=sha1')
 
