diff --git a/libgnome-desktop/gnome-bg/meson.build b/libgnome-desktop/gnome-bg/meson.build
index 76e0b1f..295e868 100644
--- a/libgnome-desktop/gnome-bg/meson.build
+++ b/libgnome-desktop/gnome-bg/meson.build
@@ -40,21 +40,21 @@ libgnome_bg = library('gnome-bg-4',
   ],
 )
 
-libgnome_bg_gir = gnome.generate_gir(libgnome_bg,
-  sources: [libgnome_bg_headers, libgnome_bg_sources],
-  export_packages: 'gnome-bg-4',
-  namespace: 'GnomeBG',
-  nsversion: '4.0',
-  includes: [libgnome_desktop_base_gir[0], 'GdkPixbuf-2.0', 'Gdk-4.0'],
-  extra_args: ['-DGNOME_DESKTOP_USE_UNSTABLE_API', '--quiet', '--warn-all'],
-  identifier_prefix: 'Gnome',
-  symbol_prefix: 'gnome',
-  install: true,
-)
+# libgnome_bg_gir = gnome.generate_gir(libgnome_bg,
+#   sources: [libgnome_bg_headers, libgnome_bg_sources],
+#   export_packages: 'gnome-bg-4',
+#   namespace: 'GnomeBG',
+#   nsversion: '4.0',
+#   includes: [ 'GdkPixbuf-2.0', 'Gdk-4.0'],
+#   extra_args: ['-DGNOME_DESKTOP_USE_UNSTABLE_API', '--quiet', '--warn-all'],
+#   identifier_prefix: 'Gnome',
+#   symbol_prefix: 'gnome',
+#   install: true,
+# )
 
 libgnome_bg_dep = declare_dependency(
   sources: [
-    libgnome_bg_gir,
+    # libgnome_bg_gir,
   ],
   dependencies: libgnome_bg_deps,
   link_with: libgnome_bg,
diff --git a/libgnome-desktop/gnome-rr/meson.build b/libgnome-desktop/gnome-rr/meson.build
index 3845d9f..bfdd774 100644
--- a/libgnome-desktop/gnome-rr/meson.build
+++ b/libgnome-desktop/gnome-rr/meson.build
@@ -44,22 +44,22 @@ libgnome_rr = library('gnome-rr-4',
   ],
 )
 
-libgnome_rr_gir = gnome.generate_gir(libgnome_rr,
-  sources: [libgnome_rr_headers, libgnome_rr_sources],
-  export_packages: 'gnome-rr-4',
-  namespace: 'GnomeRR',
-  nsversion: '4.0',
-  includes: [libgnome_desktop_base_gir[0], 'Gdk-4.0'],
-  extra_args: ['-DGNOME_DESKTOP_USE_UNSTABLE_API', '--quiet', '--warn-all'],
-  identifier_prefix: 'GnomeRR',
-  symbol_prefix: 'gnome_rr',
-  install: true,
-)
+# libgnome_rr_gir = gnome.generate_gir(libgnome_rr,
+#   sources: [libgnome_rr_headers, libgnome_rr_sources],
+#   export_packages: 'gnome-rr-4',
+#   namespace: 'GnomeRR',
+#   nsversion: '4.0',
+#   includes: ['Gdk-4.0'],
+#   extra_args: ['-DGNOME_DESKTOP_USE_UNSTABLE_API', '--quiet', '--warn-all'],
+#   identifier_prefix: 'GnomeRR',
+#   symbol_prefix: 'gnome_rr',
+#   install: true,
+# )
 
 libgnome_rr_dep = declare_dependency(
   sources: [
     dbus_xrandr_built_sources,
-    libgnome_rr_gir,
+    # libgnome_rr_gir,
   ],
   dependencies: libgnome_rr_deps,
   link_with: libgnome_rr,
diff --git a/libgnome-desktop/meson.build b/libgnome-desktop/meson.build
index 708a4ae..85415d4 100644
--- a/libgnome-desktop/meson.build
+++ b/libgnome-desktop/meson.build
@@ -154,9 +154,8 @@ if get_option('build_gtk4')
   )
 
   libgnome_desktop_base_dep = declare_dependency(
-    sources: [
-      libgnome_desktop_base_gir,
-    ],
+    # sources: [
+    # ],
     dependencies: base_deps,
     link_with: libgnome_desktop_base,
     include_directories: [
