commit 401c3d4877dabeda93e51365dc066b0514e59257
Author: Relic1212 <67271773+Relic1212@users.noreply.github.com>
Date:   Tue May 12 07:34:52 2026 +0200

    fix

diff --git a/src/main.rs b/src/main.rs
index 07461b8..8a21a3c 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -350,7 +350,7 @@ async fn start(
 	start_component("cosmic-idle", span, &process_manager, &env_vars).await;
 
 	#[cfg(feature = "autostart")]
-	if !*is_systemd_used() {
+	
 		info!("looking for autostart folders");
 		let mut directories_to_scan = Vec::new();
 
@@ -455,7 +455,7 @@ async fn start(
 			}
 		}
 		info!("started {} programs", dedupe.len());
-	}
+	
 
 	let mut sigterm = signal(SignalKind::terminate()).expect("Failed to bind SIGTERM handler");
 	let mut sigint = signal(SignalKind::interrupt()).expect("Failed to bind SIGINT handler");
