diff --git a/src/main.rs b/src/main.rs
index edb8700..2d4e904 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -352,7 +352,7 @@ async fn start(
 	start_component("cosmic-idle", span, &process_manager, &env_vars).await;
 
 	#[cfg(feature = "autostart")]
-	if !*is_systemd_used() {
+	
 		info!("looking for autostart folders");
 		let mut directories_to_scan = Vec::new();
 
@@ -455,7 +455,7 @@ async fn start(
 			}
 		}
 		info!("started {} programs", dedupe.len());
-	}
+	
 
 	let mut signals = Signals::new(vec![libc::SIGTERM, libc::SIGINT]).unwrap();
 	let mut status = Status::Exited;
