diff --git a/src/time/time_format.cpp b/src/time/time_format.cpp
index d2de59106..d7e858731 100644
--- a/src/time/time_format.cpp
+++ b/src/time/time_format.cpp
@@ -154,7 +154,12 @@ std::string formatLocalTime(const char* fmt) {
     return *compat;
   }
 
-  const auto local = current_zone()->to_local(now);
+  // const auto local = current_zone()->to_local(now);
+  std::tm date;
+  time_t tn;
+  tn = std::chrono::system_clock::to_time_t(now);
+  localtime_r(&tn, &date);
+  const auto local = std::chrono::system_clock::from_time_t(timegm(&date));
   try {
     return std::vformat(std::locale(""), normalizedFmt, std::make_format_args(local));
   } catch (...) {
