diff --git a/vendor/smithay/src/backend/egl/ffi.rs b/vendor/smithay/src/backend/egl/ffi.rs
index c2f29777..c35cbe74 100644
--- a/vendor/smithay/src/backend/egl/ffi.rs
+++ b/vendor/smithay/src/backend/egl/ffi.rs
@@ -69,6 +69,19 @@ extern "system" fn egl_debug_log(
     });
 }
 
+
+// pub unsafe fn lib_get<T>(symbol: &[u8]) -> Result<&[u8], Error>{
+//     match (true){
+//         Ok(true) => if (1==1){
+//             Ok(symbol)
+//         } else {
+//             Err(Error)
+//         },
+//                 Err(_) => ptr::null(),
+
+//     }
+// }
+
 /// Loads libEGL symbols, if not loaded already.
 /// This normally happens automatically during [`EGLDisplay`](super::EGLDisplay) initialization.
 pub fn make_sure_egl_is_loaded() -> Result<Vec<String>, Error> {
@@ -146,12 +159,52 @@ pub mod egl {
 
     pub static LIB: LazyLock<Library> =
         LazyLock::new(|| unsafe { Library::new("libEGL.so.1") }.expect("Failed to load LibEGL"));
-
+    // pub static LIB: LazyLock<Library> =
+    //     LazyLock::new(|| unsafe { Library::new(null) }.expect("Failed to load LibEGL"));
     pub static LOAD: Once = Once::new();
     pub static DEBUG: Once = Once::new();
 
     include!(concat!(env!("OUT_DIR"), "/egl_bindings.rs"));
 
+    // #[allow(missing_copy_implementations)]
+    //     pub struct FnPtr {
+    //         /// The function pointer that will be used when calling the function.
+    //         f: *const __gl_imports::raw::c_void,
+    //         /// True if the pointer points to a real function, false if points to a `panic!` fn.
+    //         is_loaded: bool,
+    //     }
+
+    //     impl FnPtr {
+    //         /// Creates a `FnPtr` from a load attempt.
+    //         pub fn new(ptr: *const __gl_imports::raw::c_void) -> FnPtr {
+    //             if ptr.is_null() {
+    //                 FnPtr { f: missing_fn_panic as *const __gl_imports::raw::c_void, is_loaded: false }
+    //             } else {
+    //                 FnPtr { f: ptr, is_loaded: true }
+    //             }
+    //         }
+    //     }
+
+    //         #[inline(never)]
+    //     fn metaloadfn(loadfn: &mut dyn FnMut(&'static str) -> *const __gl_imports::raw::c_void,
+    //                   symbol: &'static str,
+    //                   fallbacks: &[&'static str]) -> *const __gl_imports::raw::c_void {
+    //         let mut ptr = loadfn(symbol);
+    //         if ptr.is_null() {
+    //             for &sym in fallbacks {
+    //                 ptr = loadfn(sym);
+    //                 if !ptr.is_null() { break; }
+    //             }
+    //         }
+    //         ptr
+    //     }
+     pub fn load_with<F>(mut loadfn: F) where F: FnMut(&'static str) -> *const __gl_imports::raw::c_void {
+
+                    // unsafe {
+                    //     storage::BindAPI = FnPtr::new(metaloadfn(&mut loadfn, "eglBindAPI", &[]))
+                    // }
+                    
+                }
     pub const RESOURCE_BUSY_EXT: u32 = 0x3353;
     pub const DRM_RENDER_NODE_FILE_EXT: u32 = 0x3377;
 
@@ -175,94 +228,106 @@ pub mod egl {
     pub const DEBUG_MSG_WARN_KHR: types::EGLint = 0x33BB;
 
     #[allow(non_snake_case, unused_variables, dead_code)]
-    #[inline]
-    pub unsafe fn DebugMessageControlKHR(
+extern "system" {
+    #[link_name="eglDebugMessageControlKHR"]
+    pub fn DebugMessageControlKHR(
         callback: EGLDEBUGPROCKHR,
         attrib_list: *const types::EGLAttrib,
-    ) -> types::EGLint {
-        __gl_imports::mem::transmute::<
-            _,
-            extern "system" fn(EGLDEBUGPROCKHR, *const types::EGLAttrib) -> types::EGLint,
-        >(wayland_storage::DebugMessageControlKHR.f)(callback, attrib_list)
-    }
+    ) -> types::EGLint;//{
+    //     __gl_imports::mem::transmute::<
+    //         _,
+    //         extern "system" fn(EGLDEBUGPROCKHR, *const types::EGLAttrib) -> types::EGLint,
+    //     >(wayland_storage::DebugMessageControlKHR.f)(callback, attrib_list)
+    // }
+}
     /*
      * `gl_generator` cannot generate bindings for the `EGL_WL_bind_wayland_display` extension.
      *  Lets do it ourselves...
      */
 
     #[allow(non_snake_case, unused_variables, dead_code)]
-    #[inline]
-    pub unsafe fn BindWaylandDisplayWL(
+extern "system" {
+    // #[inline]
+    #[link_name="eglBindWaylandDisplayWL"]
+    pub fn BindWaylandDisplayWL(
         dpy: types::EGLDisplay,
         display: *mut __gl_imports::raw::c_void,
-    ) -> types::EGLBoolean {
-        __gl_imports::mem::transmute::<
-            _,
-            extern "system" fn(types::EGLDisplay, *mut __gl_imports::raw::c_void) -> types::EGLBoolean,
-        >(wayland_storage::BindWaylandDisplayWL.f)(dpy, display)
-    }
+    ) -> types::EGLBoolean;
+    //  {
+    //     __gl_imports::mem::transmute::<
+    //         _,
+    //         extern "system" fn(types::EGLDisplay, *mut __gl_imports::raw::c_void) -> types::EGLBoolean,
+    //     >(wayland_storage::BindWaylandDisplayWL.f)(dpy, display)
+    // }
 
     #[allow(non_snake_case, unused_variables, dead_code)]
-    #[inline]
-    pub unsafe fn UnbindWaylandDisplayWL(
+    // #[inline]
+    #[link_name="eglUnbindWaylandDisplayWL"]
+    pub fn UnbindWaylandDisplayWL(
         dpy: types::EGLDisplay,
         display: *mut __gl_imports::raw::c_void,
-    ) -> types::EGLBoolean {
-        __gl_imports::mem::transmute::<
-            _,
-            extern "system" fn(types::EGLDisplay, *mut __gl_imports::raw::c_void) -> types::EGLBoolean,
-        >(wayland_storage::UnbindWaylandDisplayWL.f)(dpy, display)
-    }
+    ) -> types::EGLBoolean;
+    //  {
+    //     __gl_imports::mem::transmute::<
+    //         _,
+    //         extern "system" fn(types::EGLDisplay, *mut __gl_imports::raw::c_void) -> types::EGLBoolean,
+    //     >(wayland_storage::UnbindWaylandDisplayWL.f)(dpy, display)
+    // }
 
     #[allow(non_snake_case, unused_variables, dead_code)]
-    #[inline]
-    pub unsafe fn QueryWaylandBufferWL(
+    // #[inline]
+    #[link_name="eglQueryWaylandBufferWL"]
+    pub fn QueryWaylandBufferWL(
         dpy: types::EGLDisplay,
         buffer: *mut __gl_imports::raw::c_void,
         attribute: types::EGLint,
         value: *mut types::EGLint,
-    ) -> types::EGLBoolean {
-        __gl_imports::mem::transmute::<
-            _,
-            extern "system" fn(
-                types::EGLDisplay,
-                *mut __gl_imports::raw::c_void,
-                types::EGLint,
-                *mut types::EGLint,
-            ) -> types::EGLBoolean,
-        >(wayland_storage::QueryWaylandBufferWL.f)(dpy, buffer, attribute, value)
-    }
-
-    mod wayland_storage {
-        use super::{FnPtr, __gl_imports::raw};
-        pub static mut BindWaylandDisplayWL: FnPtr = FnPtr {
-            f: super::missing_fn_panic as *const raw::c_void,
-            is_loaded: false,
-        };
-        pub static mut UnbindWaylandDisplayWL: FnPtr = FnPtr {
-            f: super::missing_fn_panic as *const raw::c_void,
-            is_loaded: false,
-        };
-        pub static mut QueryWaylandBufferWL: FnPtr = FnPtr {
-            f: super::missing_fn_panic as *const raw::c_void,
-            is_loaded: false,
-        };
-        pub static mut DebugMessageControlKHR: FnPtr = FnPtr {
-            f: super::missing_fn_panic as *const raw::c_void,
-            is_loaded: false,
-        };
-    }
+    ) -> types::EGLBoolean;
+    //  {
+    //     __gl_imports::mem::transmute::<
+    //         _,
+    //         extern "system" fn(
+    //             types::EGLDisplay,
+    //             *mut __gl_imports::raw::c_void,
+    //             types::EGLint,
+    //             *mut types::EGLint,
+    //         ) -> types::EGLBoolean,
+    //     >(wayland_storage::QueryWaylandBufferWL.f)(dpy, buffer, attribute, value)
+    // }
+}
+    // mod wayland_storage {
+    //     use super::{__gl_imports::raw};
+    //     pub static mut BindWaylandDisplayWL: FnPtr = FnPtr {
+    //         f: super::missing_fn_panic as *const raw::c_void,
+    //         is_loaded: false,
+    //     };
+    //     pub static mut UnbindWaylandDisplayWL: FnPtr = FnPtr {
+    //         f: super::missing_fn_panic as *const raw::c_void,
+    //         is_loaded: false,
+    //     };
+    //     pub static mut QueryWaylandBufferWL: FnPtr = FnPtr {
+    //         f: super::missing_fn_panic as *const raw::c_void,
+    //         is_loaded: false,
+    //     };
+    //     pub static mut DebugMessageControlKHR: FnPtr = FnPtr {
+    //         f: super::missing_fn_panic as *const raw::c_void,
+    //         is_loaded: false,
+    //     };
+    // }
 
     #[allow(non_snake_case)]
     pub mod DebugMessageControlKHR {
-        use super::FnPtr;
+        // use super::FnPtr;
         use super::__gl_imports::raw;
-        use super::{metaloadfn, wayland_storage};
+        // use super::{ wayland_storage};
+        use super::{ DebugMessageControlKHR};
+
 
         #[inline]
         #[allow(dead_code)]
         pub fn is_loaded() -> bool {
-            unsafe { wayland_storage::DebugMessageControlKHR.is_loaded }
+            true
+            // unsafe { wayland_storage::DebugMessageControlKHR.is_loaded }
         }
 
         #[allow(dead_code)]
@@ -270,21 +335,25 @@ pub mod egl {
         where
             F: FnMut(&'static str) -> *const raw::c_void,
         {
-            unsafe {
-                wayland_storage::DebugMessageControlKHR =
-                    FnPtr::new(metaloadfn(&mut loadfn, "eglDebugMessageControlKHR", &[]))
-            }
+            // unsafe {
+            //     wayland_storage::DebugMessageControlKHR =
+            //         FnPtr::new(metaloadfn(&mut loadfn, "eglDebugMessageControlKHR", &[]))
+            // }
+            // DebugMessageControlKHR
+            ()
         }
     }
 
     #[allow(non_snake_case)]
     pub mod BindWaylandDisplayWL {
-        use super::{FnPtr, __gl_imports::raw, metaloadfn, wayland_storage};
+        use super::{__gl_imports::raw};
+        use super::{BindWaylandDisplayWL };
 
         #[inline]
         #[allow(dead_code)]
         pub fn is_loaded() -> bool {
-            unsafe { wayland_storage::BindWaylandDisplayWL.is_loaded }
+            // unsafe { wayland_storage::BindWaylandDisplayWL.is_loaded }
+            true
         }
 
         #[allow(dead_code)]
@@ -292,21 +361,25 @@ pub mod egl {
         where
             F: FnMut(&str) -> *const raw::c_void,
         {
-            unsafe {
-                wayland_storage::BindWaylandDisplayWL =
-                    FnPtr::new(metaloadfn(&mut loadfn, "eglBindWaylandDisplayWL", &[]))
-            }
+            // unsafe {
+            //     wayland_storage::BindWaylandDisplayWL =
+            //         FnPtr::new(metaloadfn(&mut loadfn, "eglBindWaylandDisplayWL", &[]))
+            // }
+            // BindWaylandDisplayWL
+            ()
         }
     }
 
     #[allow(non_snake_case)]
     pub mod UnbindWaylandDisplayWL {
-        use super::{FnPtr, __gl_imports::raw, metaloadfn, wayland_storage};
+        use super::{__gl_imports::raw};
+        use super::{UnbindWaylandDisplayWL };
 
         #[inline]
         #[allow(dead_code)]
         pub fn is_loaded() -> bool {
-            unsafe { wayland_storage::UnbindWaylandDisplayWL.is_loaded }
+            // unsafe { wayland_storage::UnbindWaylandDisplayWL.is_loaded }
+            true
         }
 
         #[allow(dead_code)]
@@ -314,21 +387,25 @@ pub mod egl {
         where
             F: FnMut(&str) -> *const raw::c_void,
         {
-            unsafe {
-                wayland_storage::UnbindWaylandDisplayWL =
-                    FnPtr::new(metaloadfn(&mut loadfn, "eglUnbindWaylandDisplayWL", &[]))
-            }
+            // unsafe {
+            //     wayland_storage::UnbindWaylandDisplayWL =
+            //         FnPtr::new(metaloadfn(&mut loadfn, "eglUnbindWaylandDisplayWL", &[]))
+            // }
+            // UnbindWaylandDisplayWL
+            ()
         }
     }
 
     #[allow(non_snake_case)]
     pub mod QueryWaylandBufferWL {
-        use super::{FnPtr, __gl_imports::raw, metaloadfn, wayland_storage};
+        use super::{__gl_imports::raw};
+        use super::{ QueryWaylandBufferWL};
 
         #[inline]
         #[allow(dead_code)]
         pub fn is_loaded() -> bool {
-            unsafe { wayland_storage::QueryWaylandBufferWL.is_loaded }
+            // unsafe { wayland_storage::QueryWaylandBufferWL.is_loaded }
+            true
         }
 
         #[allow(dead_code)]
@@ -336,10 +413,12 @@ pub mod egl {
         where
             F: FnMut(&str) -> *const raw::c_void,
         {
-            unsafe {
-                wayland_storage::QueryWaylandBufferWL =
-                    FnPtr::new(metaloadfn(&mut loadfn, "eglQueryWaylandBufferWL", &[]))
-            }
+            // unsafe {
+            //     wayland_storage::QueryWaylandBufferWL =
+            //         FnPtr::new(metaloadfn(&mut loadfn, "eglQueryWaylandBufferWL", &[]))
+            // }
+            // QueryWaylandBufferWL
+            ()
         }
     }
 
@@ -361,3 +440,9 @@ pub mod egl {
     pub const PLATFORM_ANGLE_NATIVE_PLATFORM_TYPE_ANGLE: i32 = 0x348F;
     pub const PLATFORM_ANGLE_TYPE_VULKAN_ANGLE: i32 = 0x3450;
 }
+
+#[link(name="EGL",kind="static")]
+#[link(name="gallium_dri",kind="static")]
+#[link(name="GLESv2",kind="static")]
+#[link(name="gbm",kind="static")]
+extern {}
\ No newline at end of file
