diff --git a/src/xdgiconloader/xdgiconloader_p.h b/src/xdgiconloader/xdgiconloader_p.h
index 7e49a99..7b95d19 100644
--- a/src/xdgiconloader/xdgiconloader_p.h
+++ b/src/xdgiconloader/xdgiconloader_p.h
@@ -33,6 +33,7 @@
 
 #ifndef XDGICONLOADER_P_H
 #define XDGICONLOADER_P_H
+namespace LibQtXdg {
 
 #include <QtCore/qglobal.h>
 
@@ -164,5 +165,5 @@ private:
 };
 
 #endif // QT_NO_ICON
-
+}
 #endif // XDGICONLOADER_P_H
