diff --git a/src/wayland/wlr_layershell/surface.cpp b/src/wayland/wlr_layershell/surface.cpp
index 4ef2e32..ecdfd06 100644
--- a/src/wayland/wlr_layershell/surface.cpp
+++ b/src/wayland/wlr_layershell/surface.cpp
@@ -275,7 +275,6 @@ void LayerSurface::commit() {
 }
 
 void LayerSurface::attachPopup(QtWaylandClient::QWaylandShellSurface* popup) {
-#ifdef __FreeBSD__
 	// FreeBSD uses an alternate RTTI matching strategy by default which does
 	// not work across modules, preventing std::any from downcasting. On
 	// FreeBSD, Qt is built with a patch to expose the surface role through a
@@ -285,7 +284,6 @@ void LayerSurface::attachPopup(QtWaylandClient::QWaylandShellSurface* popup) {
 		this->get_popup(xdgPopup);
 		return;
 	}
-#endif
 	auto role = popup->surfaceRole(); // NOLINT
 	if (auto* popupRole = std::any_cast<::xdg_popup*>(&role)) {
 		this->get_popup(*popupRole);
