diff --git a/lib/wp/meson.build b/lib/wp/meson.build
index 9aacc624..54f00a85 100644
--- a/lib/wp/meson.build
+++ b/lib/wp/meson.build
@@ -124,18 +124,137 @@ wpbuildbasedirs = configure_file (
   configuration : wpbuildbasedirs_data,
 )
 
+wp_lib_sources += ['../../modules/module-default-nodes-api.c', 
+'../../modules/module-settings.c',
+ '../../modules/module-si-node.c',
+  '../../modules/module-mixer-api.c', 
+  '../../modules/module-portal-permissionstore.c',
+
+    '../../modules/module-si-standard-link.c', 
+   '../../modules/module-standard-event-source.c', 
+   '../../modules/module-file-monitor-api.c', 
+   '../../modules/module-si-audio-adapter.c', 
+   
+    '../../modules/module-log-settings.c',
+     '../../modules/module-dbus-connection.c', 
+     
+     ]
+
+dbus_connection_enums = gnome.mkenums_simple('dbus-connection-enums',
+  sources: [ '../../modules/dbus-connection-state.h' ],
+)
+wp_lib_sources += [dbus_connection_enums]
+
+
+wplua_include_dir = include_directories('../../modules/module-lua-scripting')
+wplua_lib_sources = [
+  '../../modules/module-lua-scripting/wplua/boxed.c',
+  '../../modules/module-lua-scripting/wplua/closure.c',
+  '../../modules/module-lua-scripting/wplua/object.c',
+  '../../modules/module-lua-scripting/wplua/userdata.c',
+  '../../modules/module-lua-scripting/wplua/value.c',
+  '../../modules/module-lua-scripting/wplua/wplua.c',
+]
+
+
+wplua_resources = gnome.compile_resources(
+    'wplua-resources',
+    '../../modules/module-lua-scripting/wplua/gresource.xml',
+    c_name: '_wplua',
+    extra_args: '--manual-register',
+    source_dir: meson.current_source_dir() / '..' / '..' / 'modules' / 'module-lua-scripting' / 'wplua' 
+)
+
+
+
+wp_lib_sources +=  wplua_lib_sources
+wp_lib_sources += wplua_resources 
+
+m_lua_scripting_resources = gnome.compile_resources(
+    'm-lua-scripting-resources',
+    '../../modules/module-lua-scripting/api/gresource.xml',
+    source_dir: meson.current_source_dir() / '..' / '..' / 'modules' / 'module-lua-scripting' / 'api',
+    c_name: '_m_lua_scripting_resources',
+    export: true,
+)
+
+
+# wp_lib_sources +=[
+#     '../../modules/module-lua-scripting/module.c',
+#     '../../modules/module-lua-scripting/script.c',
+#     '../../modules/module-lua-scripting/api/pod.c',
+#     '../../modules/module-lua-scripting/api/json.c',
+#     '../../modules/module-lua-scripting/api/api.c',
+#      m_lua_scripting_resources,
+#   ]
+
+
+reserve_device_interface_src = gnome.gdbus_codegen('reserve-device-interface',
+  sources: '../../modules/module-reserve-device/org.freedesktop.ReserveDevice1.xml',
+  interface_prefix : 'org.freedesktop.ReserveDevice1.',
+  namespace : 'Wp'
+)
+reserve_device_enums = gnome.mkenums_simple('reserve-device-enums',
+  sources: [ '../../modules/module-reserve-device/plugin.h', '../../modules/module-reserve-device/reserve-device.h' ],
+)
+reserve_device_includes = include_directories('../../modules/module-reserve-device')
+
+wp_lib_sources +=   [
+    '../../modules/module-reserve-device/plugin.c',
+    '../../modules/module-reserve-device/reserve-device.c',
+    '../../modules/module-reserve-device/transitions.c',
+    reserve_device_interface_src,
+    reserve_device_enums,
+  ]
+
+wplua_lib_i = static_library('wplua-1-' + wireplumber_api_version,
+  [ wplua_lib_sources, wplua_resources , wpenums_h, ],
+  install: false,
+  include_directories: [wplua_include_dir,wp_lib_include_dir],
+  dependencies : [lua_dep, giounix_dep, gobject_dep, gio_dep],
+)
+
+wplua_dep_i = declare_dependency(
+  link_with: wplua_lib_i,
+  include_directories: wplua_include_dir,
+  dependencies: [gobject_dep, gio_dep, lua_dep],
+)
+wp_lua_scripting_lib = library(
+  'wireplumber-module-lua-scripting-1',
+  [
+    '../../modules/module-lua-scripting/module.c',
+    '../../modules/module-lua-scripting/script.c',
+    '../../modules/module-lua-scripting/api/pod.c',
+    '../../modules/module-lua-scripting/api/json.c',
+    '../../modules/module-lua-scripting/api/api.c',
+     m_lua_scripting_resources,  wpenums_c, wpenums_h,
+  ],
+  install : true,
+  # install_dir : wireplumber_module_dir,
+  include_directories: [wplua_include_dir,wp_lib_include_dir],
+  
+  dependencies : [gobject_dep, gio_dep, pipewire_dep, wplua_dep_i, libintl_dep],
+)
+
+
+
 wp_lib = library('wireplumber-' + wireplumber_api_version,
   wp_lib_sources, wp_lib_priv_sources, wpenums_c, wpenums_h, wpversion, wpbuildbasedirs,
   c_args : [
     '-DBUILDING_WP',
   ],
   install: true,
-  include_directories: wp_lib_include_dir,
-  dependencies : [gobject_dep, gmodule_dep, gio_dep, pipewire_dep, libintl_dep],
+  include_directories: [ wp_lib_include_dir , include_directories('../../modules'), 
+  # include_directories('../../modules/module-lua-scripting'), 
+  reserve_device_includes],
+  dependencies : [gobject_dep, gmodule_dep, gio_dep, pipewire_dep, libintl_dep, giounix_dep , libintl_dep, mathlib,lua_dep],
   soversion: wireplumber_so_version,
   version: wireplumber_libversion,
+  # link_with: wplua_lib_i,
+  link_with: wp_lua_scripting_lib,
 )
 
+
 wp_dep = declare_dependency(
   link_with: wp_lib,
   sources: wp_gen_sources,
diff --git a/lib/wp/private/internal-comp-loader.c b/lib/wp/private/internal-comp-loader.c
index b6435335..71a33642 100644
--- a/lib/wp/private/internal-comp-loader.c
+++ b/lib/wp/private/internal-comp-loader.c
@@ -12,6 +12,20 @@
 
 #include <pipewire/impl.h>
 
+#include "module-portal-permissionstore.h"
+#include "module-mixer-api.h"
+#include "module-default-nodes-api.h"
+#include "module-lua-scripting.h"
+#include "module-file-monitor-api.h"
+#include "module-log-settings.h"
+#include "module-si-audio-adapter.h"
+#include "module-settings.h"
+#include "module-reserve-device.h"
+#include "module-standard-event-source.h"
+#include "module-si-standard-link.h"
+#include "module-si-node.h"
+#include "module-dbus-connection.h"
+
 WP_DEFINE_LOCAL_LOG_TOPIC ("wp-internal-comp-loader")
 
 /*** ComponentData ***/
@@ -789,6 +803,26 @@ static const struct {
   { "settings-instance", load_settings_instance },
 };
 
+
+static const struct {
+  const gchar * name;
+   GObject* (*load) (WpCore * core, WpSpaJson * args, GError ** error);
+} builtin_components_ii[] = {
+  { "libwireplumber-module-dbus-connection", libwireplumber_module_dbus_connection__wireplumber__module_init },
+  { "libwireplumber-module-default-nodes-api", libwireplumber_module_default_nodes_api__wireplumber__module_init },
+  { "libwireplumber-module-file-monitor-api", libwireplumber_module_file_monitor_api__wireplumber__module_init },
+  { "libwireplumber-module-log-settings", libwireplumber_module_log_settings__wireplumber__module_init},
+  { "libwireplumber-module-lua-scripting", libwireplumber_module_lua_scripting__wireplumber__module_init},
+  { "libwireplumber-module-mixer-api", libwireplumber_module_mixer_api__wireplumber__module_init},
+  { "libwireplumber-module-portal-permissionstore", libwireplumber_module_portal_permissionstore__wireplumber__module_init},
+  { "libwireplumber-module-reserve-device", libwireplumber_module_reserve_device__wireplumber__module_init },
+  { "libwireplumber-module-settings", libwireplumber_module_settings__wireplumber__module_init },
+  { "libwireplumber-module-si-audio-adapter" , libwireplumber_module_si_audio_adapter__wireplumber__module_init},
+  {"libwireplumber-module-si-node", libwireplumber_module_si_node__wireplumber__module_init},
+  { "libwireplumber-module-si-standard-link", libwireplumber_module_si_standard_link__wireplumber__module_init},
+  { "libwireplumber-module-standard-event-source", libwireplumber_module_standard_event_source__wireplumber__module_init},
+};
+
 /*** WpInternalCompLoader ***/
 
 struct _WpInternalCompLoader
@@ -838,6 +872,96 @@ load_module (WpCore * core, const gchar * module_name, WpSpaJson * args,
         "Failed to open %s: %s", module_path, g_module_error ());
     return NULL;
   }
+  // gmodule = NULL;
+
+
+
+        // if (strcmp(module_name, "libwireplumber-module-portal-permissionstore") == 0) {
+        // &module_init = &libwireplumber_module_portal_permissionstore__pipewire__module_init;
+
+        // } else 
+        
+
+        // if (strcmp(module_name, "libwireplumber-module-mixer-api") == 0) {
+        // &module_init = &libwireplumber_module_mixer_api__pipewire__module_init;
+
+        // } else 
+        
+
+        // if (strcmp(module_name, "libwireplumber-module-default-nodes-api") == 0) {
+        // &module_init = &libwireplumber_module_default_nodes_api__pipewire__module_init;
+
+        // } else 
+        
+
+        // if (strcmp(module_name, "libwireplumber-module-lua-scripting") == 0) {
+        // &module_init = &libwireplumber_module_lua_scripting__pipewire__module_init;
+
+        // } else 
+        
+
+        // if (strcmp(module_name, "libwireplumber-module-file-monitor-api") == 0) {
+        // &module_init = &libwireplumber_module_file_monitor_api__pipewire__module_init;
+
+        // } else 
+        
+
+        // if (strcmp(module_name, "libwireplumber-module-log-settings") == 0) {
+        // &module_init = &libwireplumber_module_log_settings__pipewire__module_init;
+
+        // } else 
+        
+
+        // if (strcmp(module_name, "libwireplumber-module-si-audio-adapter") == 0) {
+        // init_func = &libwireplumber_module_si_audio_adapter__pipewire__module_init;
+
+        // } else 
+        
+
+        // if (strcmp(module_name, "libwireplumber-module-settings") == 0) {
+        // &module_init = &libwireplumber_module_settings__pipewire__module_init;
+
+        // } else 
+        
+
+        // if (strcmp(module_name, "libwireplumber-module-reserve-device") == 0) {
+        // &module_init = &libwireplumber_module_reserve_device__pipewire__module_init;
+
+        // } else 
+        
+
+        // if (strcmp(module_name, "libwireplumber-module-standard-event-source") == 0) {
+        // &module_init = &libwireplumber_module_standard_event_source__pipewire__module_init;
+
+        // } else 
+        
+
+        // if (strcmp(module_name, "libwireplumber-module-si-standard-link") == 0) {
+        // &module_init = &libwireplumber_module_si_standard_link__pipewire__module_init;
+
+        // } else 
+        
+
+        // if (strcmp(module_name, "libwireplumber-module-si-node") == 0) {
+        // &module_init = &libwireplumber_module_si_node__pipewire__module_init;
+
+        // } else 
+        
+
+        // if (strcmp(module_name, "libwireplumber-module-dbus-connection") == 0) {
+        // &module_init = &libwireplumber_module_dbus_connection__pipewire__module_init;
+
+        // } else 
+        
+
+        // if (strcmp(module_name, "") == 0) {
+        // &module_init = &__pipewire__module_init;
+
+        // } else 
+        //  {
+
+        //  }
+        
 
   if (!g_module_symbol (gmodule, WP_MODULE_INIT_SYMBOL, &module_init)) {
     g_set_error (error, WP_DOMAIN_LIBRARY, WP_LIBRARY_ERROR_OPERATION_FAILED,
@@ -975,13 +1099,40 @@ wp_internal_comp_loader_load (WpComponentLoader * self, WpCore * core,
       g_autoptr (GError) error = NULL;
       g_autoptr (GObject) o = NULL;
 
-      o = load_module (core, component, args, &error);
-      if (o)
-        g_task_return_pointer (task, g_steal_pointer (&o), g_object_unref);
-      else
-        g_task_return_error (task, g_steal_pointer (&error));
+      // o = load_module (core, component, args, &error);
+      // if (o)
+      //   g_task_return_pointer (task, g_steal_pointer (&o), g_object_unref);
+      // else
+      //   g_task_return_error (task, g_steal_pointer (&error));
+
+      int found = 0;
+      for (guint i = 0; i < G_N_ELEMENTS (builtin_components_ii); i++) {
+          wp_trace_object (core, "checking if %s == %s\n", component, builtin_components_ii[i].name);
+        if (g_str_equal (component, builtin_components_ii[i].name)) {
+
+          o = ((WpModuleInitFunc) (builtin_components_ii[i].load) ) (core, args, error);
+          if (o)
+            g_task_return_pointer (task, g_steal_pointer (&o), g_object_unref);
+          else
+            g_task_return_error (task, g_steal_pointer (&error));
+          found = 1;
+          break;
+        }
+
+        
+      }
+      if (found == 0){
+         g_task_return_new_error (task, WP_DOMAIN_LIBRARY,
+          WP_LIBRARY_ERROR_INVALID_ARGUMENT,
+          "invalid 'module' component: %s", component);
+         }
+    
+
+
+
     }
-    else if (g_str_equal (type, "pw-module")) {
+    else 
+    if (g_str_equal (type, "pw-module")) {
       if (!pw_context_load_module (wp_core_get_pw_context (core), component,
               args ? wp_spa_json_get_data (args) : NULL, NULL)) {
         g_task_return_new_error (task, WP_DOMAIN_LIBRARY,
@@ -995,7 +1146,7 @@ wp_internal_comp_loader_load (WpComponentLoader * self, WpCore * core,
     else if (g_str_equal (type, "virtual")) {
       g_task_return_pointer (task, NULL, NULL);
     }
-    else if (g_str_equal (type, "built-in")) {
+    else if ((g_str_equal (type, "built-in")) || (g_str_equal (type, "module"))  ) {
       for (guint i = 0; i < G_N_ELEMENTS (builtin_components); i++) {
         if (g_str_equal (component, builtin_components[i].name)) {
           builtin_components[i].load (task, core, args);
diff --git a/meson.build b/meson.build
index e197d837..a2059247 100644
--- a/meson.build
+++ b/meson.build
@@ -161,9 +161,9 @@ common_args = [
 add_project_arguments(common_args, language: 'c')
 
 subdir('lib')
-if build_modules
-  subdir('modules')
-endif
+# if build_modules
+#   subdir('modules')
+# endif
 subdir('src')
 subdir('po')
 subdir('docs')
diff --git a/modules/.h b/modules/.h
new file mode 100644
index 00000000..b87b5cea
--- /dev/null
+++ b/modules/.h
@@ -0,0 +1,8 @@
+
+#ifndef _INCLUDED
+#define _INCLUDED
+
+GObject *
+__wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error);
+
+#endif
diff --git a/modules/meson.build b/modules/meson.build
index a3aadae3..b3990af8 100644
--- a/modules/meson.build
+++ b/modules/meson.build
@@ -1,4 +1,4 @@
-shared_library(
+library(
   'wireplumber-module-settings',
   [
     'module-settings.c',
@@ -11,7 +11,7 @@ shared_library(
 dbus_connection_enums = gnome.mkenums_simple('dbus-connection-enums',
   sources: [ 'dbus-connection-state.h' ],
 )
-shared_library(
+library(
   'wireplumber-module-dbus-connection',
   [
     'module-dbus-connection.c',
@@ -22,7 +22,7 @@ shared_library(
   dependencies : [wp_dep],
 )
 
-shared_library(
+library(
   'wireplumber-module-default-nodes-api',
   [
     'module-default-nodes-api.c',
@@ -33,7 +33,7 @@ shared_library(
 )
 
 subdir('module-reserve-device')
-shared_library(
+library(
   'wireplumber-module-reserve-device',
   [
     'module-reserve-device/plugin.c',
@@ -48,7 +48,7 @@ shared_library(
   dependencies : [wp_dep, giounix_dep],
 )
 
-shared_library(
+library(
   'wireplumber-module-portal-permissionstore',
   [
     'module-portal-permissionstore.c',
@@ -58,7 +58,7 @@ shared_library(
   dependencies : [wp_dep, giounix_dep],
 )
 
-shared_library(
+library(
   'wireplumber-module-si-audio-adapter',
   [
     'module-si-audio-adapter.c',
@@ -68,7 +68,7 @@ shared_library(
   dependencies : [wp_dep, pipewire_dep],
 )
 
-shared_library(
+library(
   'wireplumber-module-si-node',
   [
     'module-si-node.c',
@@ -78,7 +78,7 @@ shared_library(
   dependencies : [wp_dep, pipewire_dep],
 )
 
-shared_library(
+library(
   'wireplumber-module-si-standard-link',
   [
     'module-si-standard-link.c',
@@ -89,7 +89,7 @@ shared_library(
 )
 
 subdir('module-lua-scripting')
-shared_library(
+library(
   'wireplumber-module-lua-scripting',
   [
     'module-lua-scripting/module.c',
@@ -104,7 +104,7 @@ shared_library(
   dependencies : [wp_dep, pipewire_dep, wplua_dep, libintl_dep],
 )
 
-shared_library(
+library(
   'wireplumber-module-mixer-api',
   [
     'module-mixer-api.c',
@@ -114,7 +114,7 @@ shared_library(
   dependencies : [wp_dep, pipewire_dep, mathlib],
 )
 
-shared_library(
+library(
   'wireplumber-module-file-monitor-api',
   [
     'module-file-monitor-api.c',
@@ -124,7 +124,7 @@ shared_library(
   dependencies : [wp_dep, pipewire_dep],
 )
 
-shared_library(
+library(
   'wireplumber-module-log-settings',
   [
     'module-log-settings.c',
@@ -135,7 +135,7 @@ shared_library(
 )
 
 if libsystemd_dep.found() or libelogind_dep.found()
-  shared_library(
+  library(
     'wireplumber-module-logind',
     [
       'module-logind.c',
@@ -146,7 +146,7 @@ if libsystemd_dep.found() or libelogind_dep.found()
   )
 endif
 
-shared_library(
+library(
   'wireplumber-module-standard-event-source',
   [
     'module-standard-event-source.c',
diff --git a/modules/module-dbus-connection.c b/modules/module-dbus-connection.c
index b337be5e..810302c6 100644
--- a/modules/module-dbus-connection.c
+++ b/modules/module-dbus-connection.c
@@ -253,7 +253,7 @@ wp_dbus_connection_class_init (WpDBusConnectionClass * klass)
 }
 
 WP_PLUGIN_EXPORT GObject *
-wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error)
+libwireplumber_module_dbus_connection__wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error)
 {
   return G_OBJECT (g_object_new (
       wp_dbus_connection_get_type(),
diff --git a/modules/module-dbus-connection.h b/modules/module-dbus-connection.h
new file mode 100644
index 00000000..41575bed
--- /dev/null
+++ b/modules/module-dbus-connection.h
@@ -0,0 +1,8 @@
+
+#ifndef LIBWIREPLUMBER_MODULE_DBUS_CONNECTION_INCLUDED
+#define LIBWIREPLUMBER_MODULE_DBUS_CONNECTION_INCLUDED
+
+GObject *
+libwireplumber_module_dbus_connection__wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error);
+
+#endif
diff --git a/modules/module-default-nodes-api.c b/modules/module-default-nodes-api.c
index ca32b56d..e4437b66 100644
--- a/modules/module-default-nodes-api.c
+++ b/modules/module-default-nodes-api.c
@@ -309,7 +309,7 @@ wp_default_nodes_api_class_init (WpDefaultNodesApiClass * klass)
 }
 
 WP_PLUGIN_EXPORT GObject *
-wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error)
+libwireplumber_module_default_nodes_api__wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error)
 {
   return G_OBJECT (g_object_new (wp_default_nodes_api_get_type (),
       "name", "default-nodes-api",
diff --git a/modules/module-default-nodes-api.h b/modules/module-default-nodes-api.h
new file mode 100644
index 00000000..f7439373
--- /dev/null
+++ b/modules/module-default-nodes-api.h
@@ -0,0 +1,8 @@
+
+#ifndef LIBWIREPLUMBER_MODULE_DEFAULT_NODES_API_INCLUDED
+#define LIBWIREPLUMBER_MODULE_DEFAULT_NODES_API_INCLUDED
+
+GObject *
+libwireplumber_module_default_nodes_api__wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error);
+
+#endif
diff --git a/modules/module-file-monitor-api.c b/modules/module-file-monitor-api.c
index 991bb44a..83087f46 100644
--- a/modules/module-file-monitor-api.c
+++ b/modules/module-file-monitor-api.c
@@ -204,7 +204,7 @@ wp_file_monitor_api_class_init (WpFileMonitorApiClass * klass)
 }
 
 WP_PLUGIN_EXPORT GObject *
-wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error)
+libwireplumber_module_file_monitor_api__wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error)
 {
   return G_OBJECT (g_object_new (wp_file_monitor_api_get_type (),
       "name", "file-monitor-api",
diff --git a/modules/module-file-monitor-api.h b/modules/module-file-monitor-api.h
new file mode 100644
index 00000000..321a0bd0
--- /dev/null
+++ b/modules/module-file-monitor-api.h
@@ -0,0 +1,8 @@
+
+#ifndef LIBWIREPLUMBER_MODULE_FILE_MONITOR_API_INCLUDED
+#define LIBWIREPLUMBER_MODULE_FILE_MONITOR_API_INCLUDED
+
+GObject *
+libwireplumber_module_file_monitor_api__wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error);
+
+#endif
diff --git a/modules/module-log-settings.c b/modules/module-log-settings.c
index 489f7cab..fc732d2f 100644
--- a/modules/module-log-settings.c
+++ b/modules/module-log-settings.c
@@ -94,7 +94,7 @@ wp_log_settings_plugin_class_init (WpLogSettingsPluginClass * klass)
 }
 
 WP_PLUGIN_EXPORT GObject *
-wireplumber__module_init (WpCore * core, GVariant * args, GError ** error)
+libwireplumber_module_log_settings__wireplumber__module_init (WpCore * core, GVariant * args, GError ** error)
 {
   return G_OBJECT (g_object_new (wp_log_settings_plugin_get_type (),
           "name", "log-settings",
diff --git a/modules/module-log-settings.h b/modules/module-log-settings.h
new file mode 100644
index 00000000..c2b71798
--- /dev/null
+++ b/modules/module-log-settings.h
@@ -0,0 +1,8 @@
+
+#ifndef LIBWIREPLUMBER_MODULE_LOG_SETTINGS_INCLUDED
+#define LIBWIREPLUMBER_MODULE_LOG_SETTINGS_INCLUDED
+
+GObject *
+libwireplumber_module_log_settings__wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error);
+
+#endif
diff --git a/modules/module-lua-scripting.h b/modules/module-lua-scripting.h
new file mode 100644
index 00000000..7b37e21f
--- /dev/null
+++ b/modules/module-lua-scripting.h
@@ -0,0 +1,8 @@
+
+#ifndef LIBWIREPLUMBER_MODULE_LUA_SCRIPTING_INCLUDED
+#define LIBWIREPLUMBER_MODULE_LUA_SCRIPTING_INCLUDED
+
+GObject *
+libwireplumber_module_lua_scripting__wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error);
+
+#endif
diff --git a/modules/module-lua-scripting/module.c b/modules/module-lua-scripting/module.c
index df64922d..3eb37e54 100644
--- a/modules/module-lua-scripting/module.c
+++ b/modules/module-lua-scripting/module.c
@@ -218,7 +218,7 @@ wp_lua_scripting_component_loader_init (WpComponentLoaderInterface * iface)
 }
 
 WP_PLUGIN_EXPORT GObject *
-wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error)
+libwireplumber_module_lua_scripting__wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error)
 {
   return G_OBJECT (g_object_new (wp_lua_scripting_plugin_get_type (),
       "name", "lua-scripting",
diff --git a/modules/module-mixer-api.c b/modules/module-mixer-api.c
index 502640c6..666fabe2 100644
--- a/modules/module-mixer-api.c
+++ b/modules/module-mixer-api.c
@@ -625,7 +625,7 @@ wp_mixer_api_class_init (WpMixerApiClass * klass)
 }
 
 WP_PLUGIN_EXPORT GObject *
-wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error)
+libwireplumber_module_mixer_api__wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error)
 {
   return G_OBJECT (g_object_new (wp_mixer_api_get_type (),
       "name", "mixer-api",
diff --git a/modules/module-mixer-api.h b/modules/module-mixer-api.h
new file mode 100644
index 00000000..a5ed934f
--- /dev/null
+++ b/modules/module-mixer-api.h
@@ -0,0 +1,8 @@
+
+#ifndef LIBWIREPLUMBER_MODULE_MIXER_API_INCLUDED
+#define LIBWIREPLUMBER_MODULE_MIXER_API_INCLUDED
+
+GObject *
+libwireplumber_module_mixer_api__wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error);
+
+#endif
diff --git a/modules/module-portal-permissionstore.c b/modules/module-portal-permissionstore.c
index ceb7cdb9..ab572101 100644
--- a/modules/module-portal-permissionstore.c
+++ b/modules/module-portal-permissionstore.c
@@ -283,7 +283,7 @@ wp_portal_permissionstore_plugin_class_init (
 }
 
 WP_PLUGIN_EXPORT GObject *
-wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error)
+libwireplumber_module_portal_permissionstore__wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error)
 {
   return G_OBJECT (g_object_new (
       wp_portal_permissionstore_plugin_get_type(),
diff --git a/modules/module-portal-permissionstore.h b/modules/module-portal-permissionstore.h
new file mode 100644
index 00000000..7b18e954
--- /dev/null
+++ b/modules/module-portal-permissionstore.h
@@ -0,0 +1,8 @@
+
+#ifndef LIBWIREPLUMBER_MODULE_PORTAL_PERMISSIONSTORE_INCLUDED
+#define LIBWIREPLUMBER_MODULE_PORTAL_PERMISSIONSTORE_INCLUDED
+
+GObject *
+libwireplumber_module_portal_permissionstore__wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error);
+
+#endif
diff --git a/modules/module-reserve-device.h b/modules/module-reserve-device.h
new file mode 100644
index 00000000..a3474742
--- /dev/null
+++ b/modules/module-reserve-device.h
@@ -0,0 +1,8 @@
+
+#ifndef LIBWIREPLUMBER_MODULE_RESERVE_DEVICE_INCLUDED
+#define LIBWIREPLUMBER_MODULE_RESERVE_DEVICE_INCLUDED
+
+GObject *
+libwireplumber_module_reserve_device__wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error);
+
+#endif
diff --git a/modules/module-reserve-device/plugin.c b/modules/module-reserve-device/plugin.c
index e6789ffc..bc02414d 100644
--- a/modules/module-reserve-device/plugin.c
+++ b/modules/module-reserve-device/plugin.c
@@ -256,7 +256,7 @@ wp_reserve_device_plugin_class_init (WpReserveDevicePluginClass * klass)
 }
 
 WP_PLUGIN_EXPORT GObject *
-wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error)
+libwireplumber_module_reserve_device__wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error)
 {
   return G_OBJECT (g_object_new (wp_reserve_device_plugin_get_type (),
       "name", "reserve-device",
diff --git a/modules/module-settings.c b/modules/module-settings.c
index f13aa77a..59754c12 100644
--- a/modules/module-settings.c
+++ b/modules/module-settings.c
@@ -409,7 +409,7 @@ wp_settings_plugin_class_init (WpSettingsPluginClass * klass)
 }
 
 WP_PLUGIN_EXPORT GObject *
-wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error)
+libwireplumber_module_settings__wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error)
 {
   g_autofree gchar *metadata_name = NULL;
   if (args)
diff --git a/modules/module-settings.h b/modules/module-settings.h
new file mode 100644
index 00000000..b8b1cb19
--- /dev/null
+++ b/modules/module-settings.h
@@ -0,0 +1,8 @@
+
+#ifndef LIBWIREPLUMBER_MODULE_SETTINGS_INCLUDED
+#define LIBWIREPLUMBER_MODULE_SETTINGS_INCLUDED
+
+GObject *
+libwireplumber_module_settings__wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error);
+
+#endif
diff --git a/modules/module-si-audio-adapter.c b/modules/module-si-audio-adapter.c
index c6667c54..dd4aee52 100644
--- a/modules/module-si-audio-adapter.c
+++ b/modules/module-si-audio-adapter.c
@@ -808,7 +808,7 @@ si_audio_adapter_linkable_init (WpSiLinkableInterface * iface)
 }
 
 WP_PLUGIN_EXPORT GObject *
-wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error)
+libwireplumber_module_si_audio_adapter__wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error)
 {
   return G_OBJECT (wp_si_factory_new_simple (SI_FACTORY_NAME,
       si_audio_adapter_get_type ()));
diff --git a/modules/module-si-audio-adapter.h b/modules/module-si-audio-adapter.h
new file mode 100644
index 00000000..85019c60
--- /dev/null
+++ b/modules/module-si-audio-adapter.h
@@ -0,0 +1,8 @@
+
+#ifndef LIBWIREPLUMBER_MODULE_SI_AUDIO_ADAPTER_INCLUDED
+#define LIBWIREPLUMBER_MODULE_SI_AUDIO_ADAPTER_INCLUDED
+
+GObject *
+libwireplumber_module_si_audio_adapter__wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error);
+
+#endif
diff --git a/modules/module-si-node.c b/modules/module-si-node.c
index e88516ae..9da3b759 100644
--- a/modules/module-si-node.c
+++ b/modules/module-si-node.c
@@ -227,7 +227,7 @@ si_node_linkable_init (WpSiLinkableInterface * iface)
 }
 
 WP_PLUGIN_EXPORT GObject *
-wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error)
+libwireplumber_module_si_node__wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error)
 {
   return G_OBJECT (wp_si_factory_new_simple (SI_FACTORY_NAME,
       si_node_get_type ()));
diff --git a/modules/module-si-node.h b/modules/module-si-node.h
new file mode 100644
index 00000000..8de55359
--- /dev/null
+++ b/modules/module-si-node.h
@@ -0,0 +1,8 @@
+
+#ifndef LIBWIREPLUMBER_MODULE_SI_NODE_INCLUDED
+#define LIBWIREPLUMBER_MODULE_SI_NODE_INCLUDED
+
+GObject *
+libwireplumber_module_si_node__wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error);
+
+#endif
diff --git a/modules/module-si-standard-link.c b/modules/module-si-standard-link.c
index 342e00a0..3be5b2e6 100644
--- a/modules/module-si-standard-link.c
+++ b/modules/module-si-standard-link.c
@@ -764,7 +764,7 @@ si_standard_link_link_init (WpSiLinkInterface * iface)
 }
 
 WP_PLUGIN_EXPORT GObject *
-wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error)
+libwireplumber_module_si_standard_link__wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error)
 {
   return G_OBJECT (wp_si_factory_new_simple (SI_FACTORY_NAME,
       si_standard_link_get_type ()));
diff --git a/modules/module-si-standard-link.h b/modules/module-si-standard-link.h
new file mode 100644
index 00000000..eebc8bfa
--- /dev/null
+++ b/modules/module-si-standard-link.h
@@ -0,0 +1,8 @@
+
+#ifndef LIBWIREPLUMBER_MODULE_SI_STANDARD_LINK_INCLUDED
+#define LIBWIREPLUMBER_MODULE_SI_STANDARD_LINK_INCLUDED
+
+GObject *
+libwireplumber_module_si_standard_link__wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error);
+
+#endif
diff --git a/modules/module-standard-event-source.c b/modules/module-standard-event-source.c
index 6ed2a4be..f98743dd 100644
--- a/modules/module-standard-event-source.c
+++ b/modules/module-standard-event-source.c
@@ -481,7 +481,7 @@ wp_standard_event_source_class_init (WpStandardEventSourceClass * klass)
 }
 
 WP_PLUGIN_EXPORT GObject *
-wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error)
+libwireplumber_module_standard_event_source__wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error)
 {
   return G_OBJECT (g_object_new (wp_standard_event_source_get_type (),
       "name", "standard-event-source",
diff --git a/modules/module-standard-event-source.h b/modules/module-standard-event-source.h
new file mode 100644
index 00000000..85ffe71e
--- /dev/null
+++ b/modules/module-standard-event-source.h
@@ -0,0 +1,8 @@
+
+#ifndef LIBWIREPLUMBER_MODULE_STANDARD_EVENT_SOURCE_INCLUDED
+#define LIBWIREPLUMBER_MODULE_STANDARD_EVENT_SOURCE_INCLUDED
+
+GObject *
+libwireplumber_module_standard_event_source__wireplumber__module_init (WpCore * core, WpSpaJson * args, GError ** error);
+
+#endif
diff --git a/src/meson.build b/src/meson.build
index 04f42c2f..e75209aa 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -25,7 +25,7 @@ if build_daemon
 
   wp_sources = [
     'main.c',
-  ]
+  ] + m_lua_scripting_resources
 
   wireplumber = executable('wireplumber',
     wp_sources,
