diff --git a/src/vulkan/wsi/wsi_common.c b/src/vulkan/wsi/wsi_common.c
index f78e4baa22a..c0067f1ae97 100644
--- a/src/vulkan/wsi/wsi_common.c
+++ b/src/vulkan/wsi/wsi_common.c
@@ -105,7 +105,9 @@ wsi_device_init(struct wsi_device *wsi,
       .sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_PROPERTIES_2,
       .pNext = &wsi->pci_bus_info,
    };
-   GetPhysicalDeviceProperties2(pdevice, &pdp2);
+   if (GetPhysicalDeviceProperties2){
+      GetPhysicalDeviceProperties2(pdevice, &pdp2);
+   }
 
    wsi->maxImageDimension2D = pdp2.properties.limits.maxImageDimension2D;
    assert(pdp2.properties.limits.optimalBufferCopyRowPitchAlignment <= UINT32_MAX);
