diff --git a/loader/CMakeLists.txt b/loader/CMakeLists.txt
index 008b71da4..db4d1590c 100644
--- a/loader/CMakeLists.txt
+++ b/loader/CMakeLists.txt
@@ -419,7 +419,8 @@ else()
         message(WARNING "The APPLE_STATIC_LOADER option has been set. Note that this will only work on MacOS and is not supported "
                 "or tested as part of the loader. Use it at your own risk.")
     else()
-        add_library(vulkan SHARED)
+        add_library(vulkan STATIC)
+        target_compile_definitions(vulkan PRIVATE APPLE_STATIC_LOADER)
     endif()
 
     target_sources(vulkan PRIVATE ${NORMAL_LOADER_SRCS})
@@ -530,14 +531,14 @@ endif()
 # Generate CMake Configuration File (IE: VulkanLoaderConfig.cmake)
 install(TARGETS vulkan EXPORT VulkanLoaderConfig)
 set_target_properties(vulkan PROPERTIES EXPORT_NAME "Loader")
-install(EXPORT VulkanLoaderConfig DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/VulkanLoader NAMESPACE Vulkan::)
+# install(EXPORT VulkanLoaderConfig DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/VulkanLoader NAMESPACE Vulkan::)
 
-# Generate CMake Version File (IE: VulkanLoaderConfigVersion.cmake)
-include(CMakePackageConfigHelpers)
+# # Generate CMake Version File (IE: VulkanLoaderConfigVersion.cmake)
+# include(CMakePackageConfigHelpers)
 
-set(version_config "${CMAKE_CURRENT_BINARY_DIR}/generated/VulkanLoaderConfigVersion.cmake")
-write_basic_package_version_file("${version_config}" COMPATIBILITY SameMajorVersion)
-install(FILES "${version_config}" DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/VulkanLoader)
+# set(version_config "${CMAKE_CURRENT_BINARY_DIR}/generated/VulkanLoaderConfigVersion.cmake")
+# write_basic_package_version_file("${version_config}" COMPATIBILITY SameMajorVersion)
+# install(FILES "${version_config}" DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/VulkanLoader)
 
 # Generate PkgConfig File (IE: vulkan.pc)
 # NOTE: Hopefully in the future CMake can generate .pc files natively.
diff --git a/loader/vk_loader_platform.h b/loader/vk_loader_platform.h
index 987337c93..491c9a964 100644
--- a/loader/vk_loader_platform.h
+++ b/loader/vk_loader_platform.h
@@ -76,7 +76,7 @@
 #endif  // defined(_WIN32)
 
 #if defined(APPLE_STATIC_LOADER) && !defined(__APPLE__)
-#error "APPLE_STATIC_LOADER can only be defined on Apple platforms!"
+// #error "APPLE_STATIC_LOADER can only be defined on Apple platforms!"
 #endif
 
 #if defined(APPLE_STATIC_LOADER)
