diff --git a/CMakeLists.txt b/CMakeLists.txt
index 655a6d1..bdfab3e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,15 +51,21 @@ if(PipeWire_FOUND)
 endif()
 option(BUILD_CAMERAINDICATOR "Build the camera indicator Plasma applet" ${BUILD_CAMERAINDICATOR_DEFAULT})
 
+
+pkg_check_modules(XKBCOMMON xkbcommon REQUIRED IMPORTED_TARGET)
+pkg_check_modules(POLKIT_GOBJECT polkit-gobject-1 REQUIRED IMPORTED_TARGET)
+
 find_package(Qt6 ${QT_MIN_VERSION} CONFIG REQUIRED COMPONENTS
                     Concurrent Core5Compat DBus Location Network Positioning Quick QuickWidgets
-                    ShaderTools Sql Svg Widgets)
+                    ShaderTools Sql Svg Widgets TextToSpeech)
 
 if (Qt6Gui_VERSION VERSION_GREATER_EQUAL "6.10.0")
     find_package(Qt6GuiPrivate ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
     find_package(Qt6WaylandClientPrivate ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
 endif()
 
+find_package(QRencode)
+
 find_package(QCoro6 REQUIRED COMPONENTS Core)
 find_package(KF6 ${KF6_MIN_VERSION} REQUIRED COMPONENTS
                     Archive Auth CoreAddons Crash DBusAddons Declarative GlobalAccel GuiAddons
@@ -175,6 +181,14 @@ if(PLASMA_X11_DEFAULT_SESSION)
     set(WITH_X11 1)
 endif()
 
+find_package(X11)
+function(check_X11_lib _lib)
+    if (NOT TARGET X11::${_lib})
+        message(SEND_ERROR "Required component ${_lib} of X11 was not found")
+    endif()
+endfunction()
+check_X11_lib(Xtst)
+
 if(WITH_X11)
     find_package(X11)
     set_package_properties(X11 PROPERTIES DESCRIPTION "X11 libraries"
diff --git a/libnotificationmanager/CMakeLists.txt b/libnotificationmanager/CMakeLists.txt
index 0121827..3506560 100644
--- a/libnotificationmanager/CMakeLists.txt
+++ b/libnotificationmanager/CMakeLists.txt
@@ -102,7 +102,7 @@ set_target_properties(notificationmanager PROPERTIES
 ecm_add_qml_module(notificationmanager URI org.kde.notificationmanager GENERATE_PLUGIN_SOURCE)
 ecm_finalize_qml_module(notificationmanager)
 
-install(TARGETS notificationmanager EXPORT notificationmanagerLibraryTargets ${KDE_INSTALL_TARGETS_DEFAULT_ARGS} )
+install(TARGETS notificationmanager notificationmanager_resources_1 EXPORT notificationmanagerLibraryTargets ${KDE_INSTALL_TARGETS_DEFAULT_ARGS} )
 
 install(FILES
     server.h
diff --git a/libtaskmanager/CMakeLists.txt b/libtaskmanager/CMakeLists.txt
index 9e21332..e45227c 100644
--- a/libtaskmanager/CMakeLists.txt
+++ b/libtaskmanager/CMakeLists.txt
@@ -98,7 +98,7 @@ ecm_qt_declare_logging_category(taskmanager
     EXPORT PLASMAWORKSPACE
 )
 
-install(TARGETS taskmanager EXPORT libtaskmanagerLibraryTargets ${KDE_INSTALL_TARGETS_DEFAULT_ARGS} )
+install(TARGETS taskmanager taskmanager_resources_1 EXPORT libtaskmanagerLibraryTargets ${KDE_INSTALL_TARGETS_DEFAULT_ARGS} )
 
 install(FILES
     abstracttasksmodel.h
