diff --git a/src/kpackage/CMakeLists.txt b/src/kpackage/CMakeLists.txt
index 51dc149..0f2a222 100644
--- a/src/kpackage/CMakeLists.txt
+++ b/src/kpackage/CMakeLists.txt
@@ -85,9 +85,9 @@ install(FILES
         DESTINATION ${KDE_INSTALL_INCLUDEDIR_KF}/KPackage/KPackage COMPONENT Devel)
 
 install(TARGETS KF6Package EXPORT KF6PackageTargets ${KF_INSTALL_TARGETS_DEFAULT_ARGS})
-if (NOT BUILD_SHARED_LIBS)
-    install(TARGETS kpackage_common_STATIC EXPORT KF6PackageTargets ${KF_INSTALL_TARGETS_DEFAULT_ARGS})
-endif()
+# if (NOT BUILD_SHARED_LIBS)
+#     install(TARGETS kpackage_common_STATIC EXPORT KF6PackageTargets ${KF_INSTALL_TARGETS_DEFAULT_ARGS})
+# endif()
 
 if(BUILD_QCH)
     ecm_add_qch(
