diff --git a/vulkaninfo/vulkaninfo.cpp b/vulkaninfo/vulkaninfo.cpp
index 2c11226a..198949dd 100644
--- a/vulkaninfo/vulkaninfo.cpp
+++ b/vulkaninfo/vulkaninfo.cpp
@@ -200,7 +200,7 @@ void DumpSurfaceCapabilities(Printer &p, AppInstance &inst, AppGpu &gpu, AppSurf
             surface_caps2.sType = VK_STRUCTURE_TYPE_SURFACE_CAPABILITIES_2_KHR;
             surface_caps2.pNext = &SurfacePresentScalingCapabilitiesKHR;
 
-            VkResult err = vkGetPhysicalDeviceSurfaceCapabilities2KHR(gpu.phys_device, &surface_info, &surface_caps2);
+            VkResult err = dl_vkGetPhysicalDeviceSurfaceCapabilities2KHR(gpu.phys_device, &surface_info, &surface_caps2);
             if (err != VK_SUCCESS) {
                 continue;
             }
@@ -208,7 +208,7 @@ void DumpSurfaceCapabilities(Printer &p, AppInstance &inst, AppGpu &gpu, AppSurf
             std::vector<VkPresentModeKHR> compatible_present_modes{SurfacePresentModeCompatibilityKHR.presentModeCount};
             SurfacePresentModeCompatibilityKHR.pPresentModes = compatible_present_modes.data();
 
-            err = vkGetPhysicalDeviceSurfaceCapabilities2KHR(gpu.phys_device, &surface_info, &surface_caps2);
+            err = dl_vkGetPhysicalDeviceSurfaceCapabilities2KHR(gpu.phys_device, &surface_info, &surface_caps2);
 
             if (err == VK_SUCCESS) {
                 ObjectWrapper present_mode_obj(p, VkPresentModeKHRString(mode));
@@ -764,15 +764,15 @@ AppDisplayPlane::AppDisplayPlane(AppGpu &gpu, uint32_t index, const VkDisplayPla
     : global_index(index), properties(in_prop) {
     std::ostringstream display_name;
 
-    supported_displays = GetVector<VkDisplayKHR>("vkGetDisplayPlaneSupportedDisplaysKHR", vkGetDisplayPlaneSupportedDisplaysKHR,
+    supported_displays = GetVector<VkDisplayKHR>("vkGetDisplayPlaneSupportedDisplaysKHR", dl_vkGetDisplayPlaneSupportedDisplaysKHR,
                                                  gpu.phys_device, global_index);
 }
 
 std::vector<AppDisplayPlane> enumerate_display_planes(AppGpu &gpu) {
     std::vector<AppDisplayPlane> result;
-    if (vkGetPhysicalDeviceDisplayPlanePropertiesKHR) {
+    if (dl_vkGetPhysicalDeviceDisplayPlanePropertiesKHR) {
         auto planes = GetVector<VkDisplayPlanePropertiesKHR>("vkGetPhysicalDeviceDisplayPlanePropertiesKHR",
-                vkGetPhysicalDeviceDisplayPlanePropertiesKHR, gpu.phys_device);
+                dl_vkGetPhysicalDeviceDisplayPlanePropertiesKHR, gpu.phys_device);
         for (uint32_t i = 0; i < planes.size(); i++) {
             result.emplace_back(gpu, i, planes[i]);
         }
@@ -786,7 +786,7 @@ AppDisplayMode::AppDisplayMode(AppGpu &gpu, const VkDisplayModePropertiesKHR &in
     for (auto plane : supported_planes) {
         VkDisplayPlaneCapabilitiesKHR cap;
 
-        vkGetDisplayPlaneCapabilitiesKHR(gpu.phys_device, properties.displayMode, plane, &cap);
+        dl_vkGetDisplayPlaneCapabilitiesKHR(gpu.phys_device, properties.displayMode, plane, &cap);
         capabilities[plane] = cap;
     }
 }
@@ -800,7 +800,7 @@ static std::string MakeName(uint32_t index, const VkDisplayPropertiesKHR &prop)
 AppDisplay::AppDisplay(AppGpu &gpu, uint32_t index, const VkDisplayPropertiesKHR &in_properties,
                        const std::vector<AppDisplayPlane> &all_planes)
     : global_index(index), name(MakeName(index, in_properties)), properties(in_properties) {
-    auto mode_props = GetVector<VkDisplayModePropertiesKHR>("vkGetDisplayModePropertiesKHR", vkGetDisplayModePropertiesKHR,
+    auto mode_props = GetVector<VkDisplayModePropertiesKHR>("vkGetDisplayModePropertiesKHR", dl_vkGetDisplayModePropertiesKHR,
                                                             gpu.phys_device, properties.display);
 
     std::set<uint32_t> supported_planes;
@@ -820,9 +820,9 @@ AppDisplay::AppDisplay(AppGpu &gpu, uint32_t index, const VkDisplayPropertiesKHR
 std::vector<AppDisplay> enumerate_displays(AppGpu &gpu, const std::vector<AppDisplayPlane> &all_planes) {
     std::vector<AppDisplay> result;
 
-    if (vkGetPhysicalDeviceDisplayPropertiesKHR) {
+    if (dl_vkGetPhysicalDeviceDisplayPropertiesKHR) {
         auto properties = GetVector<VkDisplayPropertiesKHR>("vkGetPhysicalDeviceDisplayPropertiesKHR",
-                vkGetPhysicalDeviceDisplayPropertiesKHR, gpu.phys_device);
+                dl_vkGetPhysicalDeviceDisplayPropertiesKHR, gpu.phys_device);
 
         for (uint32_t i = 0; i < properties.size(); i++) {
             result.emplace_back(gpu, i, properties[i], all_planes);
@@ -1013,7 +1013,7 @@ void DumpGpuProfileCapabilities(Printer &p, AppGpu &gpu, bool show_promoted_stru
                             format_props2.formatProperties = formats.props;
                             std::unique_ptr<format_properties2_chain> chain_for_format_props2;
                             setup_format_properties2_chain(format_props2, chain_for_format_props2, gpu);
-                            vkGetPhysicalDeviceFormatProperties2KHR(gpu.phys_device, fmt, &format_props2);
+                            dl_vkGetPhysicalDeviceFormatProperties2KHR(gpu.phys_device, fmt, &format_props2);
                             chain_iterator_format_properties2(p, gpu, format_props2.pNext);
                         }
                         already_printed_formats.insert(fmt);
@@ -1107,7 +1107,7 @@ void DumpGpuProfileCapabilities(Printer &p, AppGpu &gpu, bool show_promoted_stru
 }
 void PrintProfileBaseInfo(Printer &p, const std::string &device_name, uint32_t apiVersion, const std::string &device_label,
                           const std::vector<std::string> &capabilities) {
-    ObjectWrapper vk_info(p, device_name);
+    ObjectWrapper dl_vk_info(p, device_name);
     p.PrintKeyValue("version", 1);
     p.PrintKeyString("api-version", APIVersion(apiVersion).str());
     p.PrintKeyString("label", device_label);
@@ -1535,7 +1535,7 @@ int main(int argc, char **argv) {
                 try {
                     // check if the surface is supported by the physical device before adding it to the list
                     VkBool32 supported = VK_FALSE;
-                    VkResult err = vkGetPhysicalDeviceSurfaceSupportKHR(gpu->phys_device, 0, surface_extension.surface, &supported);
+                    VkResult err = dl_vkGetPhysicalDeviceSurfaceSupportKHR(gpu->phys_device, 0, surface_extension.surface, &supported);
                     if (err != VK_SUCCESS || supported == VK_FALSE) continue;
 
                     surfaces.push_back(
diff --git a/vulkaninfo/vulkaninfo.h b/vulkaninfo/vulkaninfo.h
index e15583e6..205bd92c 100644
--- a/vulkaninfo/vulkaninfo.h
+++ b/vulkaninfo/vulkaninfo.h
@@ -390,7 +390,7 @@ struct AppVideoProfile {
 
         init_profile(*this);
 
-        VkResult result = vkGetPhysicalDeviceVideoCapabilitiesKHR(phys_device, &profile_info, &capabilities);
+        VkResult result = dl_vkGetPhysicalDeviceVideoCapabilitiesKHR(phys_device, &profile_info, &capabilities);
         if (result != VK_SUCCESS) {
             supported = false;
             return;
@@ -411,7 +411,7 @@ struct AppVideoProfile {
 
             uint32_t video_format_property_count = 0;
             result =
-                vkGetPhysicalDeviceVideoFormatPropertiesKHR(phys_device, &video_format_info, &video_format_property_count, nullptr);
+                dl_vkGetPhysicalDeviceVideoFormatPropertiesKHR(phys_device, &video_format_info, &video_format_property_count, nullptr);
             if (result != VK_SUCCESS) {
                 continue;
             }
@@ -424,7 +424,7 @@ struct AppVideoProfile {
                 video_format_props_chains[i] = create_format_properties_chain_info.callback(&video_format_props[i].pNext);
             }
 
-            result = vkGetPhysicalDeviceVideoFormatPropertiesKHR(phys_device, &video_format_info, &video_format_property_count,
+            result = dl_vkGetPhysicalDeviceVideoFormatPropertiesKHR(phys_device, &video_format_info, &video_format_property_count,
                                                                  video_format_props.data());
             if (result == VK_SUCCESS) {
                 for (uint32_t i = 0; i < video_format_property_count; ++i) {
@@ -565,8 +565,8 @@ struct AppInstance {
         }
 
         uint32_t instance_version = VK_API_VERSION_1_0;
-        if (vkEnumerateInstanceVersion) {
-            const VkResult err = vkEnumerateInstanceVersion(&instance_version);
+        if (dl_vkEnumerateInstanceVersion) {
+            const VkResult err = dl_vkEnumerateInstanceVersion(&instance_version);
             if (err) THROW_VK_ERR("vkEnumerateInstanceVersion", err);
         }
 
@@ -602,7 +602,7 @@ struct AppInstance {
             static_cast<uint32_t>(inst_exts.size()),
             inst_exts.data()};
 
-        VkResult err = vkCreateInstance(&inst_info, nullptr, &instance);
+        VkResult err = dl_vkCreateInstance(&inst_info, nullptr, &instance);
         if (err == VK_ERROR_INCOMPATIBLE_DRIVER) {
             std::cerr << "Cannot create " API_NAME " instance.\n";
             std::cerr << "This problem is often caused by a faulty installation of the " API_NAME
@@ -615,15 +615,15 @@ struct AppInstance {
 
         load_vulkan_instance_functions(instance);
 
-        err = vkCreateDebugReportCallbackEXT(instance, &dbg_info, nullptr, &debug_callback);
+        err = dl_vkCreateDebugReportCallbackEXT(instance, &dbg_info, nullptr, &debug_callback);
         if (err != VK_SUCCESS) {
             THROW_VK_ERR("vkCreateDebugReportCallbackEXT", err);
         }
     }
 
     ~AppInstance() {
-        if (debug_callback) vkDestroyDebugReportCallbackEXT(instance, debug_callback, nullptr);
-        if (vkDestroyInstance) vkDestroyInstance(instance, nullptr);
+        if (debug_callback) dl_vkDestroyDebugReportCallbackEXT(instance, debug_callback, nullptr);
+        if (dl_vkDestroyInstance) dl_vkDestroyInstance(instance, nullptr);
         unload_vulkan_library();
     }
 
@@ -639,7 +639,7 @@ struct AppInstance {
     void AppGetInstanceExtensions() {
         /* Scan layers */
         auto global_layer_properties =
-            GetVector<VkLayerProperties>("vkEnumerateInstanceLayerProperties", vkEnumerateInstanceLayerProperties);
+            GetVector<VkLayerProperties>("vkEnumerateInstanceLayerProperties", dl_vkEnumerateInstanceLayerProperties);
 
         for (const auto &layer : global_layer_properties) {
             global_layers.push_back(LayerExtensionList{layer, AppGetGlobalLayerExtensions(layer.layerName)});
@@ -760,16 +760,16 @@ struct AppInstance {
     void AddSurfaceExtension(SurfaceExtension ext) { surface_extensions.push_back(ext); }
 
     std::vector<VkExtensionProperties> AppGetGlobalLayerExtensions(const char *layer_name) {
-        return GetVector<VkExtensionProperties>("vkEnumerateInstanceExtensionProperties", vkEnumerateInstanceExtensionProperties,
+        return GetVector<VkExtensionProperties>("vkEnumerateInstanceExtensionProperties", dl_vkEnumerateInstanceExtensionProperties,
                                                 layer_name);
     }
 
     std::vector<VkPhysicalDevice> FindPhysicalDevices() {
-        return GetVector<VkPhysicalDevice>("vkEnumeratePhysicalDevices", vkEnumeratePhysicalDevices, instance);
+        return GetVector<VkPhysicalDevice>("vkEnumeratePhysicalDevices", dl_vkEnumeratePhysicalDevices, instance);
     }
 
     std::vector<VkExtensionProperties> AppGetPhysicalDeviceLayerExtensions(VkPhysicalDevice phys_device, const char *layer_name) {
-        return GetVector<VkExtensionProperties>("vkEnumerateDeviceExtensionProperties", vkEnumerateDeviceExtensionProperties,
+        return GetVector<VkExtensionProperties>("vkEnumerateDeviceExtensionProperties", dl_vkEnumerateDeviceExtensionProperties,
                                                 phys_device, layer_name);
     }
 };
@@ -787,7 +787,7 @@ struct AppInstance {
 //-----------------------------------------------------------
 #if defined(VULKANINFO_WSI_ENABLED)
 static void AppDestroySurface(AppInstance &inst, VkSurfaceKHR surface) {  // same for all platforms
-    vkDestroySurfaceKHR(inst.instance, surface, nullptr);
+    dl_vkDestroySurfaceKHR(inst.instance, surface, nullptr);
 }
 #endif  // defined(VULKANINFO_WSI_ENABLED)
 //-----------------------------------------------------------
@@ -854,7 +854,7 @@ static VkSurfaceKHR AppCreateWin32Surface(AppInstance &inst) {
     createInfo.hwnd = inst.h_wnd;
 
     VkSurfaceKHR surface;
-    VkResult err = vkCreateWin32SurfaceKHR(inst.instance, &createInfo, nullptr, &surface);
+    VkResult err = dl_vkCreateWin32SurfaceKHR(inst.instance, &createInfo, nullptr, &surface);
     if (err) THROW_VK_ERR("vkCreateWin32SurfaceKHR", err);
     return surface;
 }
@@ -913,7 +913,7 @@ static VkSurfaceKHR AppCreateXcbSurface(AppInstance &inst) {
     xcb_createInfo.window = inst.xcb_window;
 
     VkSurfaceKHR surface;
-    VkResult err = vkCreateXcbSurfaceKHR(inst.instance, &xcb_createInfo, nullptr, &surface);
+    VkResult err = dl_vkCreateXcbSurfaceKHR(inst.instance, &xcb_createInfo, nullptr, &surface);
     if (err) THROW_VK_ERR("vkCreateXcbSurfaceKHR", err);
     return surface;
 }
@@ -964,7 +964,7 @@ static VkSurfaceKHR AppCreateXlibSurface(AppInstance &inst) {
     createInfo.window = inst.xlib_window;
 
     VkSurfaceKHR surface;
-    VkResult err = vkCreateXlibSurfaceKHR(inst.instance, &createInfo, nullptr, &surface);
+    VkResult err = dl_vkCreateXlibSurfaceKHR(inst.instance, &createInfo, nullptr, &surface);
     if (err) THROW_VK_ERR("vkCreateXlibSurfaceKHR", err);
     return surface;
 }
@@ -993,7 +993,7 @@ static VkSurfaceKHR AppCreateMacOSSurface(AppInstance &inst) {
     createInfo.pView = inst.macos_window;
 
     VkSurfaceKHR surface;
-    VkResult err = vkCreateMacOSSurfaceMVK(inst.instance, &createInfo, nullptr, &surface);
+    VkResult err = dl_vkCreateMacOSSurfaceMVK(inst.instance, &createInfo, nullptr, &surface);
     if (err) THROW_VK_ERR("vkCreateMacOSSurfaceMVK", err);
     return surface;
 }
@@ -1019,7 +1019,7 @@ static VkSurfaceKHR AppCreateMetalSurface(AppInstance &inst) {
     createInfo.pLayer = static_cast<CAMetalLayer *>(GetCAMetalLayerFromMetalView(inst.metal_window));
 
     VkSurfaceKHR surface;
-    VkResult err = vkCreateMetalSurfaceEXT(inst.instance, &createInfo, nullptr, &surface);
+    VkResult err = dl_vkCreateMetalSurfaceEXT(inst.instance, &createInfo, nullptr, &surface);
     if (err) THROW_VK_ERR("vkCreateMetalSurfaceEXT", err);
     return surface;
 }
@@ -1058,7 +1058,7 @@ static VkSurfaceKHR AppCreateWaylandSurface(AppInstance &inst) {
     createInfo.surface = inst.wayland_surface;
 
     VkSurfaceKHR surface;
-    VkResult err = vkCreateWaylandSurfaceKHR(inst.instance, &createInfo, nullptr, &surface);
+    VkResult err = dl_vkCreateWaylandSurfaceKHR(inst.instance, &createInfo, nullptr, &surface);
     if (err) THROW_VK_ERR("vkCreateWaylandSurfaceKHR", err);
     return surface;
 }
@@ -1102,7 +1102,7 @@ static VkSurfaceKHR AppCreateDirectFBSurface(AppInstance &inst) {
     createInfo.surface = inst.directfb_surface;
 
     VkSurfaceKHR surface;
-    VkResult err = vkCreateDirectFBSurfaceEXT(inst.instance, &createInfo, nullptr, &surface);
+    VkResult err = dl_vkCreateDirectFBSurfaceEXT(inst.instance, &createInfo, nullptr, &surface);
     if (err) THROW_VK_ERR("vkCreateDirectFBSurfaceEXT", err);
     return surface;
 }
@@ -1125,7 +1125,7 @@ static VkSurfaceKHR AppCreateAndroidSurface(AppInstance &inst) {
     createInfo.window = (struct ANativeWindow *)(inst.window);
 
     VkSurfaceKHR surface;
-    VkResult err = vkCreateAndroidSurfaceKHR(inst.instance, &createInfo, NULL, &surface);
+    VkResult err = dl_vkCreateAndroidSurfaceKHR(inst.instance, &createInfo, NULL, &surface);
     if (err) THROW_VK_ERR("vkCreateAndroidSurfaceKHR", err);
     return surface;
 }
@@ -1143,7 +1143,7 @@ static VkSurfaceKHR AppCreateGgpSurface(AppInstance &inst) {
     createInfo.streamDescriptor = 1;
 
     VkSurfaceKHR surface;
-    VkResult err = vkCreateStreamDescriptorSurfaceGGP(inst.instance, &createInfo, NULL, &surface);
+    VkResult err = dl_vkCreateStreamDescriptorSurfaceGGP(inst.instance, &createInfo, NULL, &surface);
     if (err) THROW_VK_ERR("vkCreateStreamDescriptorSurfaceGGP", err);
     return surface;
 }
@@ -1179,7 +1179,7 @@ static VkSurfaceKHR AppCreateScreenSurface(AppInstance &inst) {
     createInfo.window = inst.window;
 
     VkSurfaceKHR surface;
-    VkResult err = vkCreateScreenSurfaceQNX(inst.instance, &createInfo, nullptr, &surface);
+    VkResult err = dl_vkCreateScreenSurfaceQNX(inst.instance, &createInfo, nullptr, &surface);
     if (err) THROW_VK_ERR("vkCreateScreenSurfaceQNX", err);
     return surface;
 }
@@ -1359,7 +1359,7 @@ class AppSurface {
     AppSurface(AppInstance &inst, AppGpu &gpu, VkPhysicalDevice phys_device, SurfaceExtension surface_extension)
         : inst(inst), phys_device(phys_device), surface_extension(surface_extension) {
         surf_present_modes =
-            GetVector<VkPresentModeKHR>("vkGetPhysicalDeviceSurfacePresentModesKHR", vkGetPhysicalDeviceSurfacePresentModesKHR,
+            GetVector<VkPresentModeKHR>("vkGetPhysicalDeviceSurfacePresentModesKHR", dl_vkGetPhysicalDeviceSurfacePresentModesKHR,
                                         phys_device, surface_extension.surface);
 
         if (inst.CheckExtensionEnabled(VK_KHR_GET_SURFACE_CAPABILITIES_2_EXTENSION_NAME)) {
@@ -1368,15 +1368,15 @@ class AppSurface {
             VkSurfaceFormat2KHR init{};
             init.sType = VK_STRUCTURE_TYPE_SURFACE_FORMAT_2_KHR;
             surf_formats2 = GetVectorInit<VkSurfaceFormat2KHR>(
-                "vkGetPhysicalDeviceSurfaceFormats2KHR", vkGetPhysicalDeviceSurfaceFormats2KHR, init, phys_device, &surface_info2);
+                "vkGetPhysicalDeviceSurfaceFormats2KHR", dl_vkGetPhysicalDeviceSurfaceFormats2KHR, init, phys_device, &surface_info2);
         } else {
             surf_formats =
-                GetVector<VkSurfaceFormatKHR>("vkGetPhysicalDeviceSurfaceFormatsKHR", vkGetPhysicalDeviceSurfaceFormatsKHR,
+                GetVector<VkSurfaceFormatKHR>("vkGetPhysicalDeviceSurfaceFormatsKHR", dl_vkGetPhysicalDeviceSurfaceFormatsKHR,
                                               phys_device, surface_extension.surface);
         }
 
         if (inst.CheckExtensionEnabled(VK_KHR_SURFACE_EXTENSION_NAME)) {
-            VkResult err = vkGetPhysicalDeviceSurfaceCapabilitiesKHR(phys_device, surface_extension.surface, &surface_capabilities);
+            VkResult err = dl_vkGetPhysicalDeviceSurfaceCapabilitiesKHR(phys_device, surface_extension.surface, &surface_capabilities);
             if (err) THROW_VK_ERR("vkGetPhysicalDeviceSurfaceCapabilitiesKHR", err);
         }
 
@@ -1394,7 +1394,7 @@ class AppSurface {
 
             surface_info.pNext = static_cast<void *>(&win32_fullscreen_exclusive_info);
 #endif  // defined(WIN32)
-            VkResult err = vkGetPhysicalDeviceSurfaceCapabilities2KHR(phys_device, &surface_info, &surface_capabilities2_khr);
+            VkResult err = dl_vkGetPhysicalDeviceSurfaceCapabilities2KHR(phys_device, &surface_info, &surface_capabilities2_khr);
             if (err) THROW_VK_ERR("vkGetPhysicalDeviceSurfaceCapabilities2KHR", err);
         }
 
@@ -1402,7 +1402,7 @@ class AppSurface {
             surface_capabilities2_ext.sType = VK_STRUCTURE_TYPE_SURFACE_CAPABILITIES_2_EXT;
             surface_capabilities2_ext.pNext = nullptr;
             VkResult err =
-                vkGetPhysicalDeviceSurfaceCapabilities2EXT(phys_device, surface_extension.surface, &surface_capabilities2_ext);
+                dl_vkGetPhysicalDeviceSurfaceCapabilities2EXT(phys_device, surface_extension.surface, &surface_capabilities2_ext);
             if (err) THROW_VK_ERR("vkGetPhysicalDeviceSurfaceCapabilities2EXT", err);
         }
     }
@@ -1418,7 +1418,7 @@ std::vector<VkPhysicalDeviceGroupProperties> GetGroups(AppInstance &inst) {
         VkPhysicalDeviceGroupProperties group_props{};
         group_props.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_GROUP_PROPERTIES;
         return GetVectorInit<VkPhysicalDeviceGroupProperties>("vkEnumeratePhysicalDeviceGroupsKHR",
-                                                              vkEnumeratePhysicalDeviceGroupsKHR, group_props, inst.instance);
+                                                              dl_vkEnumeratePhysicalDeviceGroupsKHR, group_props, inst.instance);
     }
     return {};
 }
@@ -1427,7 +1427,7 @@ std::vector<VkPhysicalDeviceProperties> GetGroupProps(AppInstance &inst, VkPhysi
     std::vector<VkPhysicalDeviceProperties> props(group.physicalDeviceCount);
 
     for (uint32_t i = 0; i < group.physicalDeviceCount; ++i) {
-        vkGetPhysicalDeviceProperties(group.physicalDevices[i], &props[i]);
+        dl_vkGetPhysicalDeviceProperties(group.physicalDevices[i], &props[i]);
     }
 
     return props;
@@ -1459,11 +1459,11 @@ util::vulkaninfo_optional<VkDeviceGroupPresentCapabilitiesKHR> GetGroupCapabilit
 
     VkDevice logical_device = VK_NULL_HANDLE;
 
-    VkResult err = vkCreateDevice(group.physicalDevices[0], &device_ci, nullptr, &logical_device);
+    VkResult err = dl_vkCreateDevice(group.physicalDevices[0], &device_ci, nullptr, &logical_device);
     if (err != VK_SUCCESS && err != VK_ERROR_EXTENSION_NOT_PRESENT) THROW_VK_ERR("vkCreateDevice", err);
 
     if (err == VK_ERROR_EXTENSION_NOT_PRESENT) {
-        vkDestroyDevice(logical_device, nullptr);
+        dl_vkDestroyDevice(logical_device, nullptr);
         return {};
     }
 
@@ -1471,10 +1471,10 @@ util::vulkaninfo_optional<VkDeviceGroupPresentCapabilitiesKHR> GetGroupCapabilit
 
     // If the KHR_device_group extension is present, write the capabilities of the logical device into a struct for later
     // output to user.
-    err = vkGetDeviceGroupPresentCapabilitiesKHR(logical_device, &group_capabilities);
+    err = dl_vkGetDeviceGroupPresentCapabilitiesKHR(logical_device, &group_capabilities);
     if (err) THROW_VK_ERR("vkGetDeviceGroupPresentCapabilitiesKHR", err);
 
-    vkDestroyDevice(logical_device, nullptr);
+    dl_vkDestroyDevice(logical_device, nullptr);
 
     return {group_capabilities};
 }
@@ -1521,7 +1521,7 @@ VkImageCreateInfo GetImageCreateInfo(VkImageCreateFlags flags, VkFormat format,
 util::vulkaninfo_optional<ImageTypeSupport> FillImageTypeSupport(AppInstance &inst, VkPhysicalDevice phys_device, VkDevice device,
                                                                  ImageTypeSupport::Type img_type, VkImageCreateInfo image_ci) {
     VkImageFormatProperties img_props;
-    VkResult res = vkGetPhysicalDeviceImageFormatProperties(phys_device, image_ci.format, image_ci.imageType, image_ci.tiling,
+    VkResult res = dl_vkGetPhysicalDeviceImageFormatProperties(phys_device, image_ci.format, image_ci.imageType, image_ci.tiling,
                                                             image_ci.usage, image_ci.flags, &img_props);
 
     if (res == VK_SUCCESS) {
@@ -1529,14 +1529,14 @@ util::vulkaninfo_optional<ImageTypeSupport> FillImageTypeSupport(AppInstance &in
         img_type_support.type = img_type;
 
         VkImage dummy_img;
-        res = vkCreateImage(device, &image_ci, nullptr, &dummy_img);
+        res = dl_vkCreateImage(device, &image_ci, nullptr, &dummy_img);
         if (res) THROW_VK_ERR("vkCreateImage", res);
 
         VkMemoryRequirements mem_req;
-        vkGetImageMemoryRequirements(device, dummy_img, &mem_req);
+        dl_vkGetImageMemoryRequirements(device, dummy_img, &mem_req);
         img_type_support.memoryTypeBits = mem_req.memoryTypeBits;
 
-        vkDestroyImage(device, dummy_img, nullptr);
+        dl_vkDestroyImage(device, dummy_img, nullptr);
         return img_type_support;
     } else if (res == VK_ERROR_FORMAT_NOT_SUPPORTED) {
         return {};  // return empty util::vulkaninfo_optional
@@ -1570,7 +1570,7 @@ struct AppQueueFamilyProperties {
         : props(family_properties), queue_index(queue_index), pNext(pNext) {
         for (const auto &surface_ext : inst.surface_extensions) {
             present_support.push_back({surface_ext.name, VK_FALSE});
-            VkResult err = vkGetPhysicalDeviceSurfaceSupportKHR(physical_device, queue_index, surface_ext.surface,
+            VkResult err = dl_vkGetPhysicalDeviceSurfaceSupportKHR(physical_device, queue_index, surface_ext.surface,
                                                                 &present_support.back().second);
             if (err) THROW_VK_ERR("vkGetPhysicalDeviceSurfaceSupportKHR", err);
             if (present_support.back().second) {
@@ -1638,53 +1638,53 @@ struct AppGpu {
 
     AppGpu(AppInstance &inst, uint32_t id, VkPhysicalDevice phys_device, bool show_promoted_structs)
         : inst(inst), id(id), phys_device(phys_device) {
-        vkGetPhysicalDeviceProperties(phys_device, &props);
+        dl_vkGetPhysicalDeviceProperties(phys_device, &props);
 
         // needs to find the minimum of the instance and device version, and use that to print the device info
         api_version = APIVersion(props.apiVersion);
 
         device_extensions = inst.AppGetPhysicalDeviceLayerExtensions(phys_device, nullptr);
 
-        vkGetPhysicalDeviceMemoryProperties(phys_device, &memory_props);
+        dl_vkGetPhysicalDeviceMemoryProperties(phys_device, &memory_props);
 
-        vkGetPhysicalDeviceFeatures(phys_device, &features);
+        dl_vkGetPhysicalDeviceFeatures(phys_device, &features);
 
         uint32_t queue_count = 0;
-        vkGetPhysicalDeviceQueueFamilyProperties(phys_device, &queue_count, nullptr);
+        dl_vkGetPhysicalDeviceQueueFamilyProperties(phys_device, &queue_count, nullptr);
         queue_props.resize(queue_count);
-        vkGetPhysicalDeviceQueueFamilyProperties(phys_device, &queue_count, queue_props.data());
+        dl_vkGetPhysicalDeviceQueueFamilyProperties(phys_device, &queue_count, queue_props.data());
 
         if (inst.CheckExtensionEnabled(VK_KHR_GET_PHYSICAL_DEVICE_PROPERTIES_2_EXTENSION_NAME)) {
             // VkPhysicalDeviceProperties2
             props2.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_PROPERTIES_2_KHR;
             setup_phys_device_props2_chain(props2, chain_for_phys_device_props2, inst, *this, show_promoted_structs);
 
-            vkGetPhysicalDeviceProperties2KHR(phys_device, &props2);
+            dl_vkGetPhysicalDeviceProperties2KHR(phys_device, &props2);
             prepare_phys_device_props2_twocall_chain_vectors(chain_for_phys_device_props2);
-            vkGetPhysicalDeviceProperties2KHR(phys_device, &props2);
+            dl_vkGetPhysicalDeviceProperties2KHR(phys_device, &props2);
 
             // VkPhysicalDeviceMemoryProperties2
             memory_props2.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_MEMORY_PROPERTIES_2_KHR;
             setup_phys_device_mem_props2_chain(memory_props2, chain_for_phys_device_mem_props2, *this);
 
-            vkGetPhysicalDeviceMemoryProperties2KHR(phys_device, &memory_props2);
+            dl_vkGetPhysicalDeviceMemoryProperties2KHR(phys_device, &memory_props2);
 
             // VkPhysicalDeviceFeatures2
             features2.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_FEATURES_2_KHR;
             setup_phys_device_features2_chain(features2, chain_for_phys_device_features2, *this, show_promoted_structs);
 
-            vkGetPhysicalDeviceFeatures2KHR(phys_device, &features2);
+            dl_vkGetPhysicalDeviceFeatures2KHR(phys_device, &features2);
 
             // std::vector<VkPhysicalDeviceQueueFamilyProperties2>
             uint32_t queue_prop2_count = 0;
-            vkGetPhysicalDeviceQueueFamilyProperties2KHR(phys_device, &queue_prop2_count, nullptr);
+            dl_vkGetPhysicalDeviceQueueFamilyProperties2KHR(phys_device, &queue_prop2_count, nullptr);
             queue_props2.resize(queue_prop2_count);
             chain_for_queue_props2.resize(queue_prop2_count);
             for (size_t i = 0; i < queue_props2.size(); i++) {
                 queue_props2[i].sType = VK_STRUCTURE_TYPE_QUEUE_FAMILY_PROPERTIES_2_KHR;
                 setup_queue_properties2_chain(queue_props2[i], chain_for_queue_props2[i], *this);
             }
-            vkGetPhysicalDeviceQueueFamilyProperties2KHR(phys_device, &queue_prop2_count, queue_props2.data());
+            dl_vkGetPhysicalDeviceQueueFamilyProperties2KHR(phys_device, &queue_prop2_count, queue_props2.data());
 
             if (CheckPhysicalDeviceExtensionIncluded(VK_KHR_DRIVER_PROPERTIES_EXTENSION_NAME) ||
                 api_version >= VK_API_VERSION_1_2) {
@@ -1767,7 +1767,7 @@ struct AppGpu {
         device_ci.ppEnabledExtensionNames = extensions_to_enable.data();
         device_ci.pEnabledFeatures = &enabled_features;
 
-        VkResult err = vkCreateDevice(phys_device, &device_ci, nullptr, &dev);
+        VkResult err = dl_vkCreateDevice(phys_device, &device_ci, nullptr, &dev);
         if (err) THROW_VK_ERR("vkCreateDevice", err);
 
         const std::array<VkImageTiling, 2> tilings = {VK_IMAGE_TILING_OPTIMAL, VK_IMAGE_TILING_LINEAR};
@@ -1784,7 +1784,7 @@ struct AppGpu {
                 image_type_format_info.format = format;
 
                 VkFormatProperties fmt_props;
-                vkGetPhysicalDeviceFormatProperties(phys_device, format, &fmt_props);
+                dl_vkGetPhysicalDeviceFormatProperties(phys_device, format, &fmt_props);
                 if ((tiling == VK_IMAGE_TILING_OPTIMAL && fmt_props.optimalTilingFeatures == 0) ||
                     (tiling == VK_IMAGE_TILING_LINEAR && fmt_props.linearTilingFeatures == 0)) {
                     continue;
@@ -1867,10 +1867,10 @@ struct AppGpu {
         display_planes = enumerate_display_planes(*this);
         displays = enumerate_displays(*this, display_planes);
 
-        vkDestroyDevice(dev, nullptr);
+        dl_vkDestroyDevice(dev, nullptr);
         dev = VK_NULL_HANDLE;
     }
-    ~AppGpu() { vkDestroyDevice(dev, nullptr); }
+    ~AppGpu() { dl_vkDestroyDevice(dev, nullptr); }
 
     AppGpu(const AppGpu &) = delete;
     const AppGpu &operator=(const AppGpu &) = delete;
@@ -1941,35 +1941,35 @@ struct AppGpu {
 };
 
 std::vector<VkPhysicalDeviceToolPropertiesEXT> GetToolingInfo(AppGpu &gpu) {
-    if (vkGetPhysicalDeviceToolPropertiesEXT == nullptr) return {};
+    if (dl_vkGetPhysicalDeviceToolPropertiesEXT == nullptr) return {};
     return GetVector<VkPhysicalDeviceToolPropertiesEXT>("vkGetPhysicalDeviceToolPropertiesEXT",
-                                                        vkGetPhysicalDeviceToolPropertiesEXT, gpu.phys_device);
+                                                        dl_vkGetPhysicalDeviceToolPropertiesEXT, gpu.phys_device);
 }
 
 std::vector<VkCooperativeMatrixPropertiesKHR> GetCooperativeMatrixInfo(AppGpu &gpu) {
-    if (vkGetPhysicalDeviceCooperativeMatrixPropertiesKHR == nullptr) return {};
+    if (dl_vkGetPhysicalDeviceCooperativeMatrixPropertiesKHR == nullptr) return {};
     return GetVector<VkCooperativeMatrixPropertiesKHR>("vkGetPhysicalDeviceCooperativeMatrixPropertiesKHR",
-                                                       vkGetPhysicalDeviceCooperativeMatrixPropertiesKHR, gpu.phys_device);
+                                                       dl_vkGetPhysicalDeviceCooperativeMatrixPropertiesKHR, gpu.phys_device);
 }
 std::vector<VkTimeDomainKHR> GetTimeDomainInfo(AppGpu &gpu) {
-    if (vkGetPhysicalDeviceCalibrateableTimeDomainsKHR == nullptr) return {};
+    if (dl_vkGetPhysicalDeviceCalibrateableTimeDomainsKHR == nullptr) return {};
     return GetVector<VkTimeDomainKHR>("vkGetPhysicalDeviceCalibrateableTimeDomainsKHR",
-                                      vkGetPhysicalDeviceCalibrateableTimeDomainsKHR, gpu.phys_device);
+                                      dl_vkGetPhysicalDeviceCalibrateableTimeDomainsKHR, gpu.phys_device);
 }
 std::vector<VkPhysicalDeviceFragmentShadingRateKHR> GetFragmentShadingRateInfo(AppGpu &gpu) {
-    if (vkGetPhysicalDeviceFragmentShadingRatesKHR == nullptr) return {};
+    if (dl_vkGetPhysicalDeviceFragmentShadingRatesKHR == nullptr) return {};
     return GetVector<VkPhysicalDeviceFragmentShadingRateKHR>("vkGetPhysicalDeviceFragmentShadingRatesKHR",
-                                                             vkGetPhysicalDeviceFragmentShadingRatesKHR, gpu.phys_device);
+                                                             dl_vkGetPhysicalDeviceFragmentShadingRatesKHR, gpu.phys_device);
 }
 // Returns vector where each index maps to VkSampleCountFlagBits
 std::vector<VkMultisamplePropertiesEXT> GetSampleLocationInfo(AppGpu &gpu) {
-    if (vkGetPhysicalDeviceMultisamplePropertiesEXT == nullptr) return {};
+    if (dl_vkGetPhysicalDeviceMultisamplePropertiesEXT == nullptr) return {};
     std::vector<VkMultisamplePropertiesEXT> result;
     // 7 covers VK_SAMPLE_COUNT_1_BIT to 64_BIT
     for (uint32_t i = 0; i < 7; i++) {
         const VkSampleCountFlagBits sample_count = (VkSampleCountFlagBits)(1 << i);
         VkMultisamplePropertiesEXT props = {VK_STRUCTURE_TYPE_MULTISAMPLE_PROPERTIES_EXT, nullptr};
-        vkGetPhysicalDeviceMultisamplePropertiesEXT(gpu.phys_device, sample_count, &props);
+        dl_vkGetPhysicalDeviceMultisamplePropertiesEXT(gpu.phys_device, sample_count, &props);
         result.emplace_back(props);
     }
     return result;
@@ -1995,7 +1995,7 @@ struct PropFlags {
 
 PropFlags get_format_properties(const AppGpu &gpu, VkFormat fmt) {
     VkFormatProperties props;
-    vkGetPhysicalDeviceFormatProperties(gpu.phys_device, fmt, &props);
+    dl_vkGetPhysicalDeviceFormatProperties(gpu.phys_device, fmt, &props);
 
     VkFormatProperties3 props3{};
     props3.sType = VK_STRUCTURE_TYPE_FORMAT_PROPERTIES_3;
@@ -2006,7 +2006,7 @@ PropFlags get_format_properties(const AppGpu &gpu, VkFormat fmt) {
         props2.sType = VK_STRUCTURE_TYPE_FORMAT_PROPERTIES_2;
         props2.formatProperties = props;
         props2.pNext = static_cast<void *>(&props3);
-        vkGetPhysicalDeviceFormatProperties2KHR(gpu.phys_device, fmt, &props2);
+        dl_vkGetPhysicalDeviceFormatProperties2KHR(gpu.phys_device, fmt, &props2);
     }
     return {props, props3};
 }
diff --git a/vulkaninfo/vulkaninfo_functions.h b/vulkaninfo/vulkaninfo_functions.h
index 2876155c..0c0d637b 100644
--- a/vulkaninfo/vulkaninfo_functions.h
+++ b/vulkaninfo/vulkaninfo_functions.h
@@ -40,94 +40,95 @@
 void* vulkan_library;
 
 // Global Functions
-PFN_vkGetInstanceProcAddr vkGetInstanceProcAddr;
-PFN_vkEnumerateInstanceExtensionProperties vkEnumerateInstanceExtensionProperties;
-PFN_vkEnumerateInstanceLayerProperties vkEnumerateInstanceLayerProperties;
-PFN_vkEnumerateInstanceVersion vkEnumerateInstanceVersion;
-PFN_vkCreateInstance vkCreateInstance;
+__attribute__((weak)) PFN_vkGetInstanceProcAddr dl_vkGetInstanceProcAddr = NULL;
+__attribute__((weak)) PFN_vkEnumerateInstanceExtensionProperties dl_vkEnumerateInstanceExtensionProperties;
+__attribute__((weak)) PFN_vkEnumerateInstanceLayerProperties dl_vkEnumerateInstanceLayerProperties;
+__attribute__((weak)) PFN_vkEnumerateInstanceVersion dl_vkEnumerateInstanceVersion;
+__attribute__((weak)) PFN_vkCreateInstance dl_vkCreateInstance;
 
 // Per-platform Instance functions
 #if defined(VK_USE_PLATFORM_WIN32_KHR)
-PFN_vkCreateWin32SurfaceKHR vkCreateWin32SurfaceKHR;
+__attribute__((weak)) PFN_vkCreateWin32SurfaceKHR dl_vkCreateWin32SurfaceKHR;
 #endif
 #if defined(VK_USE_PLATFORM_WAYLAND_KHR)
-PFN_vkCreateWaylandSurfaceKHR vkCreateWaylandSurfaceKHR;
+__attribute__((weak)) PFN_vkCreateWaylandSurfaceKHR dl_vkCreateWaylandSurfaceKHR;
 #endif
 #if defined(VK_USE_PLATFORM_ANDROID_KHR)
-PFN_vkCreateAndroidSurfaceKHR vkCreateAndroidSurfaceKHR;
+__attribute__((weak)) PFN_vkCreateAndroidSurfaceKHR dl_vkCreateAndroidSurfaceKHR;
 #endif
 #if defined(VK_USE_PLATFORM_XLIB_KHR)
-PFN_vkCreateXlibSurfaceKHR vkCreateXlibSurfaceKHR;
+__attribute__((weak)) PFN_vkCreateXlibSurfaceKHR dl_vkCreateXlibSurfaceKHR;
 #endif
 #if defined(VK_USE_PLATFORM_XCB_KHR)
-PFN_vkCreateXcbSurfaceKHR vkCreateXcbSurfaceKHR;
+__attribute__((weak)) PFN_vkCreateXcbSurfaceKHR dl_vkCreateXcbSurfaceKHR;
 #endif
 #if defined(VK_USE_PLATFORM_METAL_EXT)
-PFN_vkCreateMetalSurfaceEXT vkCreateMetalSurfaceEXT;
+__attribute__((weak)) PFN_vkCreateMetalSurfaceEXT dl_vkCreateMetalSurfaceEXT;
 #endif
 #if defined(VK_USE_PLATFORM_MACOS_MVK)
-PFN_vkCreateMacOSSurfaceMVK vkCreateMacOSSurfaceMVK;
+__attribute__((weak)) PFN_vkCreateMacOSSurfaceMVK dl_vkCreateMacOSSurfaceMVK;
 #endif
 #if defined(VK_USE_PLATFORM_SCREEN_QNX)
-PFN_vkCreateScreenSurfaceQNX vkCreateScreenSurfaceQNX;
+__attribute__((weak)) PFN_vkCreateScreenSurfaceQNX dl_vkCreateScreenSurfaceQNX;
 #endif
 #if defined(VK_USE_PLATFORM_DISPLAY)
-PFN_vkCreateDisplayPlaneSurfaceKHR vkCreateDisplayPlaneSurfaceKHR;
+__attribute__((weak)) PFN_vkCreateDisplayPlaneSurfaceKHR dl_vkCreateDisplayPlaneSurfaceKHR;
 #endif
 #if defined(VK_USE_PLATFORM_DIRECTFB_EXT)
-PFN_vkCreateDirectFBSurfaceEXT vkCreateDirectFBSurfaceEXT;
+__attribute__((weak)) PFN_vkCreateDirectFBSurfaceEXT dl_vkCreateDirectFBSurfaceEXT;
 #endif
 #if defined(VK_USE_PLATFORM_GGP)
-PFN_vkCreateStreamDescriptorSurfaceGGP vkCreateStreamDescriptorSurfaceGGP;
+__attribute__((weak)) PFN_vkCreateStreamDescriptorSurfaceGGP dl_vkCreateStreamDescriptorSurfaceGGP;
 #endif
 // Instance functions
-PFN_vkCreateDebugReportCallbackEXT vkCreateDebugReportCallbackEXT;
-PFN_vkCreateDevice vkCreateDevice;
-PFN_vkDestroyDebugReportCallbackEXT vkDestroyDebugReportCallbackEXT;
-PFN_vkDestroyInstance vkDestroyInstance;
-PFN_vkDestroySurfaceKHR vkDestroySurfaceKHR;
-PFN_vkEnumerateDeviceExtensionProperties vkEnumerateDeviceExtensionProperties;
-PFN_vkGetDisplayModePropertiesKHR vkGetDisplayModePropertiesKHR;
-PFN_vkGetDisplayPlaneCapabilitiesKHR vkGetDisplayPlaneCapabilitiesKHR;
-PFN_vkGetDisplayPlaneSupportedDisplaysKHR vkGetDisplayPlaneSupportedDisplaysKHR;
-PFN_vkGetPhysicalDeviceDisplayPlanePropertiesKHR vkGetPhysicalDeviceDisplayPlanePropertiesKHR;
-PFN_vkGetPhysicalDeviceDisplayPropertiesKHR vkGetPhysicalDeviceDisplayPropertiesKHR;
-PFN_vkGetPhysicalDeviceFeatures vkGetPhysicalDeviceFeatures;
-PFN_vkGetPhysicalDeviceFormatProperties vkGetPhysicalDeviceFormatProperties;
-PFN_vkGetPhysicalDeviceMemoryProperties vkGetPhysicalDeviceMemoryProperties;
-PFN_vkGetPhysicalDeviceProperties vkGetPhysicalDeviceProperties;
-PFN_vkGetPhysicalDeviceQueueFamilyProperties vkGetPhysicalDeviceQueueFamilyProperties;
-PFN_vkGetPhysicalDeviceSurfaceCapabilitiesKHR vkGetPhysicalDeviceSurfaceCapabilitiesKHR;
-PFN_vkGetPhysicalDeviceSurfaceFormatsKHR vkGetPhysicalDeviceSurfaceFormatsKHR;
-PFN_vkGetPhysicalDeviceSurfacePresentModesKHR vkGetPhysicalDeviceSurfacePresentModesKHR;
-PFN_vkGetPhysicalDeviceSurfaceSupportKHR vkGetPhysicalDeviceSurfaceSupportKHR;
-PFN_vkGetPhysicalDeviceVideoCapabilitiesKHR vkGetPhysicalDeviceVideoCapabilitiesKHR;
-PFN_vkGetPhysicalDeviceSurfaceFormats2KHR vkGetPhysicalDeviceSurfaceFormats2KHR;
-PFN_vkGetPhysicalDeviceSurfaceCapabilities2KHR vkGetPhysicalDeviceSurfaceCapabilities2KHR;
-PFN_vkGetPhysicalDeviceVideoFormatPropertiesKHR vkGetPhysicalDeviceVideoFormatPropertiesKHR;
-PFN_vkEnumeratePhysicalDeviceGroupsKHR vkEnumeratePhysicalDeviceGroupsKHR;
-PFN_vkGetDeviceGroupPresentCapabilitiesKHR vkGetDeviceGroupPresentCapabilitiesKHR;
-PFN_vkGetPhysicalDeviceImageFormatProperties vkGetPhysicalDeviceImageFormatProperties;
-PFN_vkGetPhysicalDeviceProperties2KHR vkGetPhysicalDeviceProperties2KHR;
-PFN_vkGetPhysicalDeviceMemoryProperties2KHR vkGetPhysicalDeviceMemoryProperties2KHR;
-PFN_vkGetPhysicalDeviceFeatures2KHR vkGetPhysicalDeviceFeatures2KHR;
-PFN_vkGetPhysicalDeviceQueueFamilyProperties2KHR vkGetPhysicalDeviceQueueFamilyProperties2KHR;
-PFN_vkEnumeratePhysicalDevices vkEnumeratePhysicalDevices;
-PFN_vkGetPhysicalDeviceSurfaceCapabilities2EXT vkGetPhysicalDeviceSurfaceCapabilities2EXT;
-PFN_vkGetPhysicalDeviceToolPropertiesEXT vkGetPhysicalDeviceToolPropertiesEXT;
-PFN_vkGetPhysicalDeviceFormatProperties2KHR vkGetPhysicalDeviceFormatProperties2KHR;
-PFN_vkGetPhysicalDeviceCooperativeMatrixPropertiesKHR vkGetPhysicalDeviceCooperativeMatrixPropertiesKHR;
-PFN_vkGetPhysicalDeviceCalibrateableTimeDomainsKHR vkGetPhysicalDeviceCalibrateableTimeDomainsKHR;
-PFN_vkGetPhysicalDeviceFragmentShadingRatesKHR vkGetPhysicalDeviceFragmentShadingRatesKHR;
-PFN_vkGetPhysicalDeviceMultisamplePropertiesEXT vkGetPhysicalDeviceMultisamplePropertiesEXT;
+__attribute__((weak)) PFN_vkCreateDebugReportCallbackEXT dl_vkCreateDebugReportCallbackEXT;
+__attribute__((weak)) PFN_vkCreateDevice dl_vkCreateDevice;
+__attribute__((weak)) PFN_vkDestroyDebugReportCallbackEXT dl_vkDestroyDebugReportCallbackEXT;
+__attribute__((weak)) PFN_vkDestroyInstance dl_vkDestroyInstance;
+__attribute__((weak)) PFN_vkDestroySurfaceKHR dl_vkDestroySurfaceKHR;
+__attribute__((weak)) PFN_vkEnumerateDeviceExtensionProperties dl_vkEnumerateDeviceExtensionProperties;
+__attribute__((weak)) PFN_vkGetDisplayModePropertiesKHR dl_vkGetDisplayModePropertiesKHR;
+__attribute__((weak)) PFN_vkGetDisplayPlaneCapabilitiesKHR dl_vkGetDisplayPlaneCapabilitiesKHR;
+__attribute__((weak)) PFN_vkGetDisplayPlaneSupportedDisplaysKHR dl_vkGetDisplayPlaneSupportedDisplaysKHR;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceDisplayPlanePropertiesKHR dl_vkGetPhysicalDeviceDisplayPlanePropertiesKHR;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceDisplayPropertiesKHR dl_vkGetPhysicalDeviceDisplayPropertiesKHR;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceFeatures dl_vkGetPhysicalDeviceFeatures;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceFormatProperties dl_vkGetPhysicalDeviceFormatProperties;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceMemoryProperties dl_vkGetPhysicalDeviceMemoryProperties;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceProperties dl_vkGetPhysicalDeviceProperties;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceQueueFamilyProperties dl_vkGetPhysicalDeviceQueueFamilyProperties;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceSurfaceCapabilitiesKHR dl_vkGetPhysicalDeviceSurfaceCapabilitiesKHR;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceSurfaceFormatsKHR dl_vkGetPhysicalDeviceSurfaceFormatsKHR;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceSurfacePresentModesKHR dl_vkGetPhysicalDeviceSurfacePresentModesKHR;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceSurfaceSupportKHR dl_vkGetPhysicalDeviceSurfaceSupportKHR;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceVideoCapabilitiesKHR dl_vkGetPhysicalDeviceVideoCapabilitiesKHR;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceSurfaceFormats2KHR dl_vkGetPhysicalDeviceSurfaceFormats2KHR;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceSurfaceCapabilities2KHR dl_vkGetPhysicalDeviceSurfaceCapabilities2KHR;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceVideoFormatPropertiesKHR dl_vkGetPhysicalDeviceVideoFormatPropertiesKHR;
+__attribute__((weak)) PFN_vkEnumeratePhysicalDeviceGroupsKHR dl_vkEnumeratePhysicalDeviceGroupsKHR;
+__attribute__((weak)) PFN_vkGetDeviceGroupPresentCapabilitiesKHR dl_vkGetDeviceGroupPresentCapabilitiesKHR;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceImageFormatProperties dl_vkGetPhysicalDeviceImageFormatProperties;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceProperties2KHR dl_vkGetPhysicalDeviceProperties2KHR;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceMemoryProperties2KHR dl_vkGetPhysicalDeviceMemoryProperties2KHR;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceFeatures2KHR dl_vkGetPhysicalDeviceFeatures2KHR;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceQueueFamilyProperties2KHR dl_vkGetPhysicalDeviceQueueFamilyProperties2KHR;
+__attribute__((weak)) PFN_vkEnumeratePhysicalDevices dl_vkEnumeratePhysicalDevices;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceSurfaceCapabilities2EXT dl_vkGetPhysicalDeviceSurfaceCapabilities2EXT;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceToolPropertiesEXT dl_vkGetPhysicalDeviceToolPropertiesEXT;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceFormatProperties2KHR dl_vkGetPhysicalDeviceFormatProperties2KHR;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceCooperativeMatrixPropertiesKHR dl_vkGetPhysicalDeviceCooperativeMatrixPropertiesKHR;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceCalibrateableTimeDomainsKHR dl_vkGetPhysicalDeviceCalibrateableTimeDomainsKHR;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceFragmentShadingRatesKHR dl_vkGetPhysicalDeviceFragmentShadingRatesKHR;
+__attribute__((weak)) PFN_vkGetPhysicalDeviceMultisamplePropertiesEXT dl_vkGetPhysicalDeviceMultisamplePropertiesEXT;
 
 // Device functions
-PFN_vkCreateImage vkCreateImage;
-PFN_vkGetImageMemoryRequirements vkGetImageMemoryRequirements;
-PFN_vkDestroyImage vkDestroyImage;
-PFN_vkDestroyDevice vkDestroyDevice;
+__attribute__((weak)) PFN_vkCreateImage dl_vkCreateImage;
+__attribute__((weak)) PFN_vkGetImageMemoryRequirements dl_vkGetImageMemoryRequirements;
+__attribute__((weak)) PFN_vkDestroyImage dl_vkDestroyImage;
+__attribute__((weak)) PFN_vkDestroyDevice dl_vkDestroyDevice;
 
-#define LOAD_INSTANCE_FUNCTION(instance, function) function = (PFN_##function)vkGetInstanceProcAddr(instance, #function)
+#define LOAD_INSTANCE_FUNCTION(instance, function) dl_##function = (PFN_##function)dl_vkGetInstanceProcAddr(instance, #function)
+#define LOAD_INSTANCE_FUNCTION_DL(instance, function) dl_##function = (PFN_##function)dlsym(NULL, #function)
 
 static VkResult load_vulkan_library() {
     vulkan_library = NULL;
@@ -135,7 +136,7 @@ static VkResult load_vulkan_library() {
     HMODULE library = LoadLibraryA("vulkan-1.dll");
     if (library == NULL) return VK_ERROR_INITIALIZATION_FAILED;
 
-    vkGetInstanceProcAddr = (PFN_vkGetInstanceProcAddr)(void (*)(void))GetProcAddress(library, "vkGetInstanceProcAddr");
+    dl_vkGetInstanceProcAddr = (PFN_vkGetInstanceProcAddr)(void (*)(void))GetProcAddress(library, "vkGetInstanceProcAddr");
 #elif defined(__APPLE__)
     void* library = dlopen("libvulkan.dylib", RTLD_NOW | RTLD_LOCAL);
     if (library == NULL) {
@@ -151,23 +152,26 @@ static VkResult load_vulkan_library() {
         return VK_ERROR_INITIALIZATION_FAILED;
     }
 
-    vkGetInstanceProcAddr = (PFN_vkGetInstanceProcAddr)dlsym(library, "vkGetInstanceProcAddr");
+    dl_vkGetInstanceProcAddr = (PFN_vkGetInstanceProcAddr)dlsym(library, "vkGetInstanceProcAddr");
 #else  // Assume everything else supports dlopen
     void* library = dlopen("libvulkan.so.1", RTLD_NOW | RTLD_LOCAL);
     if (library == NULL) library = dlopen("libvulkan.so", RTLD_NOW | RTLD_LOCAL);
     if (library == NULL) {
         return VK_ERROR_INITIALIZATION_FAILED;
     }
-    vkGetInstanceProcAddr = (PFN_vkGetInstanceProcAddr)dlsym(library, "vkGetInstanceProcAddr");
+    void* dl_vkGetInstanceProcAddr_void = dlsym(library, "vkGetInstanceProcAddr");
+    if (dl_vkGetInstanceProcAddr == NULL){
+        dl_vkGetInstanceProcAddr = (PFN_vkGetInstanceProcAddr)dl_vkGetInstanceProcAddr_void;
+    }
 #endif
-    if (vkGetInstanceProcAddr == NULL) {
+    if (dl_vkGetInstanceProcAddr == NULL) {
         return VK_ERROR_INITIALIZATION_FAILED;
     }
     vulkan_library = library;
-    LOAD_INSTANCE_FUNCTION(NULL, vkEnumerateInstanceExtensionProperties);
-    LOAD_INSTANCE_FUNCTION(NULL, vkEnumerateInstanceLayerProperties);
-    LOAD_INSTANCE_FUNCTION(NULL, vkEnumerateInstanceVersion);
-    LOAD_INSTANCE_FUNCTION(NULL, vkCreateInstance);
+    LOAD_INSTANCE_FUNCTION_DL(NULL, vkEnumerateInstanceExtensionProperties);
+    LOAD_INSTANCE_FUNCTION_DL(NULL, vkEnumerateInstanceLayerProperties);
+    LOAD_INSTANCE_FUNCTION_DL(NULL, vkEnumerateInstanceVersion);
+    LOAD_INSTANCE_FUNCTION_DL(NULL, vkCreateInstance);
 
     return VK_SUCCESS;
 }
