diff --git a/src/login/pam_elogind.c b/src/login/pam_elogind.c
index 453bb90e5..e1c653309 100644
--- a/src/login/pam_elogind.c
+++ b/src/login/pam_elogind.c
@@ -15,6 +15,8 @@
 #include <sys/types.h>
 #include <unistd.h>
 
+#include <dlfcn.h>
+
 #include "alloc-util.h"
 #include "audit-util.h"
 #include "bus-common-errors.h"
@@ -53,6 +55,15 @@
 
 #define LOGIN_SLOW_BUS_CALL_TIMEOUT_USEC (2*USEC_PER_MINUTE)
 
+void* _pam_elogind_dlopen(const char *mod_path);
+void*  _pam_elogind_dlsym(void *handle, const char *symbol);
+/* Session Management API's */
+int pam_elogind_pam_sm_open_session(pam_handle_t *pamh, int flags,
+			int argc, const char **argv);
+
+int pam_elogind_pam_sm_close_session(pam_handle_t *pamh, int flags,
+			int argc, const char **argv);
+
 static int parse_caps(
                 pam_handle_t *handle,
                 const char *value,
@@ -894,10 +905,12 @@ static int create_session_message(
         return 0;
 }
 
-_public_ PAM_EXTERN int pam_sm_open_session(
+_public_ PAM_EXTERN int pam_elogind_pam_sm_open_session(
                 pam_handle_t *handle,
                 int flags,
                 int argc, const char **argv) {
+        
+        fprintf(stderr, "pam_elogind_pam_sm_open_session callded\n");
 
         /* Let's release the D-Bus connection once this function exits, after all the session might live
          * quite a long time, and we are not going to process the bus connection in that time, so let's
@@ -1225,7 +1238,7 @@ success:
         return apply_user_record_settings(handle, ur, debug, default_capability_bounding_set, default_capability_ambient_set);
 }
 
-_public_ PAM_EXTERN int pam_sm_close_session(
+_public_ PAM_EXTERN int pam_elogind_pam_sm_close_session(
                 pam_handle_t *handle,
                 int flags,
                 int argc, const char **argv) {
@@ -1280,3 +1293,31 @@ _public_ PAM_EXTERN int pam_sm_close_session(
 
         return PAM_SUCCESS;
 }
+
+
+void* _pam_elogind_dlopen(const char *mod_path)
+{
+        fprintf(stderr, "(pam_elogind.a) _pam_elogind_dlopen called with mod_path \"%s\"\n",mod_path);
+
+	return (void*)mod_path;
+}
+
+void*  _pam_elogind_dlsym(void *handle, const char *symbol)
+{       
+        fprintf(stderr, "(pam_elogind.a) _pam_elogind_dlsym called with symbol \"%s\"\n", symbol);
+        if (strcmp(symbol, "pam_sm_open_session" )==0){
+                fprintf(stderr, "(pam_elogind.a) returning pam_elogind_pam_sm_open_session\n");
+
+                return pam_elogind_pam_sm_open_session;
+        } 
+
+        if (strcmp(symbol, "pam_sm_close_session" )==0){
+                fprintf(stderr, "(pam_elogind.a) returning pam_elogind_pam_sm_close_session\n");
+
+                return pam_elogind_pam_sm_close_session;
+        }
+        fprintf(stderr, "(pam_elogind.a)WARNING: returning NULL\n");
+
+	return NULL;
+}
+
