diff --git a/src/dispatch_common.c b/src/dispatch_common.c
index 153eb7c..7ec5dc6 100644
--- a/src/dispatch_common.c
+++ b/src/dispatch_common.c
@@ -264,6 +264,8 @@ struct api {
     long begin_count;
 };
 
+void *epoxy_conservative_egl_dlsym___static(const char *s);
+
 static struct api api = {
 #ifndef _WIN32
     .mutex = PTHREAD_MUTEX_INITIALIZER,
@@ -308,8 +310,15 @@ get_dlopen_handle(void **handle, const char *lib_name, bool exit_on_fail, bool l
         int flags = RTLD_LAZY | RTLD_LOCAL;
         if (!load)
             flags |= RTLD_NOLOAD;
-
+        
+        fprintf(stderr, "lib_name = %s\n", lib_name);
+        if (strcmp(lib_name,EGL_LIB) != 0){
         *handle = dlopen(lib_name, flags);
+        }
+        else {
+            *handle = (void*) lib_name;
+
+        }
         if (!*handle) {
             if (exit_on_fail) {
                 fprintf(stderr, "Couldn't open %s: %s\n", lib_name, dlerror());
@@ -346,6 +355,28 @@ do_dlsym(void **handle, const char *name, bool exit_on_fail)
     return result;
 }
 
+static void *
+do_dlsym_egl(void **handle, const char *name, bool exit_on_fail)
+{
+    void *result;
+    const char *error = "";
+
+
+    result = epoxy_conservative_egl_dlsym___static(name);
+    if (!result){
+        result = dlsym(NULL,name);
+
+    }
+    if (!result)
+        error = dlerror();
+    if (!result && exit_on_fail) {
+        fprintf(stderr, "%s() not found: %s\n", name, error);
+        abort();
+    }
+
+    return result;
+}
+
 /**
  * @brief Checks whether we're using OpenGL or OpenGL ES
  *
@@ -589,7 +620,9 @@ epoxy_current_context_is_glx(void)
     }
 
 #if PLATFORM_HAS_EGL
-    sym = epoxy_conservative_egl_dlsym("eglGetCurrentContext", false);
+    sym = eglGetCurrentContext;
+    // return true;
+    // sym = epoxy_conservative_egl_dlsym("eglGetCurrentContext", false);
     if (sym) {
         if (epoxy_egl_get_current_gl_context_api() != EGL_NONE)
             return false;
@@ -638,12 +671,333 @@ epoxy_load_egl(bool exit_if_fails, bool load)
 #endif
 }
 
+void *epoxy_conservative_egl_dlsym___static(const char *s)
+{
+    if (strcmp(s, "eglBindAPI") == 0)
+    {
+        fprintf(stderr, "found sym eglBindAPI\n");
+        return eglBindAPI;
+    }
+    else
+
+        if (strcmp(s, "eglBindTexImage") == 0)
+    {
+        fprintf(stderr, "found sym eglBindTexImage\n");
+        return eglBindTexImage;
+    }
+    else
+
+        if (strcmp(s, "eglChooseConfig") == 0)
+    {
+        fprintf(stderr, "found sym eglChooseConfig\n");
+        return eglChooseConfig;
+    }
+    else
+
+        if (strcmp(s, "eglClientWaitSync") == 0)
+    {
+        fprintf(stderr, "found sym eglClientWaitSync\n");
+        return eglClientWaitSync;
+    }
+    else
+
+        if (strcmp(s, "eglCopyBuffers") == 0)
+    {
+        fprintf(stderr, "found sym eglCopyBuffers\n");
+        return eglCopyBuffers;
+    }
+    else
+
+        if (strcmp(s, "eglCreateContext") == 0)
+    {
+        fprintf(stderr, "found sym eglCreateContext\n");
+        return eglCreateContext;
+    }
+    else
+
+        if (strcmp(s, "eglCreateImage") == 0)
+    {
+        fprintf(stderr, "found sym eglCreateImage\n");
+        return eglCreateImage;
+    }
+    else
+
+        if (strcmp(s, "eglCreatePbufferFromClientBuffer") == 0)
+    {
+        fprintf(stderr, "found sym eglCreatePbufferFromClientBuffer\n");
+        return eglCreatePbufferFromClientBuffer;
+    }
+    else
+
+        if (strcmp(s, "eglCreatePbufferSurface") == 0)
+    {
+        fprintf(stderr, "found sym eglCreatePbufferSurface\n");
+        return eglCreatePbufferSurface;
+    }
+    else
+
+        if (strcmp(s, "eglCreatePixmapSurface") == 0)
+    {
+        fprintf(stderr, "found sym eglCreatePixmapSurface\n");
+        return eglCreatePixmapSurface;
+    }
+    else
+
+        if (strcmp(s, "eglCreatePlatformPixmapSurface") == 0)
+    {
+        fprintf(stderr, "found sym eglCreatePlatformPixmapSurface\n");
+        return eglCreatePlatformPixmapSurface;
+    }
+    else
+
+        if (strcmp(s, "eglCreatePlatformWindowSurface") == 0)
+    {
+        fprintf(stderr, "found sym eglCreatePlatformWindowSurface\n");
+        return eglCreatePlatformWindowSurface;
+    }
+    else
+
+        if (strcmp(s, "eglCreateSync") == 0)
+    {
+        fprintf(stderr, "found sym eglCreateSync\n");
+        return eglCreateSync;
+    }
+    else
+
+        if (strcmp(s, "eglCreateWindowSurface") == 0)
+    {
+        fprintf(stderr, "found sym eglCreateWindowSurface\n");
+        return eglCreateWindowSurface;
+    }
+    else
+
+        if (strcmp(s, "eglDestroyContext") == 0)
+    {
+        fprintf(stderr, "found sym eglDestroyContext\n");
+        return eglDestroyContext;
+    }
+    else
+
+        if (strcmp(s, "eglDestroyImage") == 0)
+    {
+        fprintf(stderr, "found sym eglDestroyImage\n");
+        return eglDestroyImage;
+    }
+    else
+
+        if (strcmp(s, "eglDestroySurface") == 0)
+    {
+        fprintf(stderr, "found sym eglDestroySurface\n");
+        return eglDestroySurface;
+    }
+    else
+
+        if (strcmp(s, "eglDestroySync") == 0)
+    {
+        fprintf(stderr, "found sym eglDestroySync\n");
+        return eglDestroySync;
+    }
+    else
+
+        if (strcmp(s, "eglGetConfigAttrib") == 0)
+    {
+        fprintf(stderr, "found sym eglGetConfigAttrib\n");
+        return eglGetConfigAttrib;
+    }
+    else
+
+        if (strcmp(s, "eglGetConfigs") == 0)
+    {
+        fprintf(stderr, "found sym eglGetConfigs\n");
+        return eglGetConfigs;
+    }
+    else
+
+        if (strcmp(s, "eglGetCurrentContext") == 0)
+    {
+        fprintf(stderr, "found sym eglGetCurrentContext\n");
+        return eglGetCurrentContext;
+    }
+    else
+
+        if (strcmp(s, "eglGetCurrentDisplay") == 0)
+    {
+        fprintf(stderr, "found sym eglGetCurrentDisplay\n");
+        return eglGetCurrentDisplay;
+    }
+    else
+
+        if (strcmp(s, "eglGetCurrentSurface") == 0)
+    {
+        fprintf(stderr, "found sym eglGetCurrentSurface\n");
+        return eglGetCurrentSurface;
+    }
+    else
+
+        if (strcmp(s, "eglGetDisplay") == 0)
+    {
+        fprintf(stderr, "found sym eglGetDisplay\n");
+        return eglGetDisplay;
+    }
+    else
+
+        if (strcmp(s, "eglGetError") == 0)
+    {
+        fprintf(stderr, "found sym eglGetError\n");
+        return eglGetError;
+    }
+    else
+
+        if (strcmp(s, "eglGetPlatformDisplay") == 0)
+    {
+        fprintf(stderr, "found sym eglGetPlatformDisplay\n");
+        return eglGetPlatformDisplay;
+    }
+    else
+
+        if (strcmp(s, "eglGetProcAddress") == 0)
+    {
+        fprintf(stderr, "found sym eglGetProcAddress\n");
+        return eglGetProcAddress;
+    }
+    else
+
+        if (strcmp(s, "eglGetSyncAttrib") == 0)
+    {
+        fprintf(stderr, "found sym eglGetSyncAttrib\n");
+        return eglGetSyncAttrib;
+    }
+    else
+
+        if (strcmp(s, "eglInitialize") == 0)
+    {
+        fprintf(stderr, "found sym eglInitialize\n");
+        return eglInitialize;
+    }
+    else
+
+        if (strcmp(s, "eglMakeCurrent") == 0)
+    {
+        fprintf(stderr, "found sym eglMakeCurrent\n");
+        return eglMakeCurrent;
+    }
+    else
+
+        if (strcmp(s, "eglQueryAPI") == 0)
+    {
+        fprintf(stderr, "found sym eglQueryAPI\n");
+        return eglQueryAPI;
+    }
+    else
+
+        if (strcmp(s, "eglQueryContext") == 0)
+    {
+        fprintf(stderr, "found sym eglQueryContext\n");
+        return eglQueryContext;
+    }
+    else
+
+        if (strcmp(s, "eglQueryString") == 0)
+    {
+        fprintf(stderr, "found sym eglQueryString\n");
+        return eglQueryString;
+    }
+    else
+
+        if (strcmp(s, "eglQuerySurface") == 0)
+    {
+        fprintf(stderr, "found sym eglQuerySurface\n");
+        return eglQuerySurface;
+    }
+    else
+
+        if (strcmp(s, "eglReleaseTexImage") == 0)
+    {
+        fprintf(stderr, "found sym eglReleaseTexImage\n");
+        return eglReleaseTexImage;
+    }
+    else
+
+        if (strcmp(s, "eglReleaseThread") == 0)
+    {
+        fprintf(stderr, "found sym eglReleaseThread\n");
+        return eglReleaseThread;
+    }
+    else
+
+        if (strcmp(s, "eglSurfaceAttrib") == 0)
+    {
+        fprintf(stderr, "found sym eglSurfaceAttrib\n");
+        return eglSurfaceAttrib;
+    }
+    else
+
+        if (strcmp(s, "eglSwapBuffers") == 0)
+    {
+        fprintf(stderr, "found sym eglSwapBuffers\n");
+        return eglSwapBuffers;
+    }
+    else
+
+        if (strcmp(s, "eglSwapInterval") == 0)
+    {
+        fprintf(stderr, "found sym eglSwapInterval\n");
+        return eglSwapInterval;
+    }
+    else
+
+        if (strcmp(s, "eglTerminate") == 0)
+    {
+        fprintf(stderr, "found sym eglTerminate\n");
+        return eglTerminate;
+    }
+    else
+
+        if (strcmp(s, "eglWaitClient") == 0)
+    {
+        fprintf(stderr, "found sym eglWaitClient\n");
+        return eglWaitClient;
+    }
+    else
+
+        if (strcmp(s, "eglWaitGL") == 0)
+    {
+        fprintf(stderr, "found sym eglWaitGL\n");
+        return eglWaitGL;
+    }
+    else
+
+        if (strcmp(s, "eglWaitNative") == 0)
+    {
+        fprintf(stderr, "found sym eglWaitNative\n");
+        return eglWaitNative;
+    }
+    else
+
+        if (strcmp(s, "eglWaitSync") == 0)
+    {
+        fprintf(stderr, "found sym eglWaitSync\n");
+        return eglWaitSync;
+    }
+    fprintf(stderr, "returning NULL\n");
+
+    return NULL;
+}
 void *
 epoxy_conservative_egl_dlsym(const char *name, bool exit_if_fails)
 {
 #if PLATFORM_HAS_EGL
+   
     if (epoxy_load_egl(exit_if_fails, exit_if_fails))
-        return do_dlsym(&api.egl_handle, name, exit_if_fails);
+    {
+        // return do_dlsym(&api.egl_handle, name, exit_if_fails);
+        void* sym = do_dlsym_egl(&api.egl_handle, name, exit_if_fails);
+        if (sym!=NULL){
+            return sym;
+        } else {
+            fprintf(stderr,"epoxy_conservative_egl_dlsym___static returned null, name=%s\n",name);
+        }
+    }
 #endif
     return NULL;
 }
diff --git a/test/egl_without_glx.c b/test/egl_without_glx.c
index 195ef41..e81471a 100644
--- a/test/egl_without_glx.c
+++ b/test/egl_without_glx.c
@@ -49,28 +49,28 @@
  * libraries we're trying to simulate not being installed on the
  * system.
  */
-void *
-dlopen(const char *filename, int flag)
-{
-    void * (*dlopen_unwrapped)(const char *filename, int flag);
-
-    if (filename) {
-        if (!strcmp(filename, "libGL.so.1"))
-            return NULL;
-#if GLES_VERSION == 2
-        if (!strcmp(filename, "libGLESv1_CM.so.1"))
-            return NULL;
-#else
-        if (!strcmp(filename, "libGLESv2.so.2"))
-            return NULL;
-#endif
-    }
-
-    dlopen_unwrapped = dlsym(RTLD_NEXT, "dlopen");
-    assert(dlopen_unwrapped);
-
-    return dlopen_unwrapped(filename, flag);
-}
+// void *
+// dlopen(const char *filename, int flag)
+// {
+//     void * (*dlopen_unwrapped)(const char *filename, int flag);
+
+//     if (filename) {
+//         if (!strcmp(filename, "libGL.so.1"))
+//             return NULL;
+// #if GLES_VERSION == 2
+//         if (!strcmp(filename, "libGLESv1_CM.so.1"))
+//             return NULL;
+// #else
+//         if (!strcmp(filename, "libGLESv2.so.2"))
+//             return NULL;
+// #endif
+//     }
+
+//     dlopen_unwrapped = dlsym(RTLD_NEXT, "dlopen");
+//     assert(dlopen_unwrapped);
+
+//     return dlopen_unwrapped(filename, flag);
+// }
 
 
 static EGLenum last_api;
@@ -83,18 +83,18 @@ static EGLenum extra_error = EGL_SUCCESS;
 static EGLBoolean
 override_eglBindAPI(EGLenum api)
 {
-    void *egl = dlopen("libEGL.so.1", RTLD_LAZY | RTLD_LOCAL);
-    EGLBoolean (*real_eglBindAPI)(EGLenum api) = dlsym(egl, "eglBindAPI");
+    // // void *egl = dlopen("libEGL.so.1", RTLD_LAZY | RTLD_LOCAL);
+    // EGLBoolean (*real_eglBindAPI)(EGLenum api) = dlsym(egl, "eglBindAPI");
 
-    last_api = api;
+    // last_api = api;
 
-    if (api == EGL_OPENGL_API) {
-        extra_error = EGL_BAD_PARAMETER;
-        return EGL_FALSE;
-    }
+    // if (api == EGL_OPENGL_API) {
+    //     extra_error = EGL_BAD_PARAMETER;
+    //     return EGL_FALSE;
+    // }
 
-    assert(real_eglBindAPI);
-    return real_eglBindAPI(api);
+    // assert(real_eglBindAPI);
+    return eglBindAPI(api);
 }
 
 /**
@@ -104,17 +104,17 @@ override_eglBindAPI(EGLenum api)
 static EGLint
 override_eglGetError(void)
 {
-    void *egl = dlopen("libEGL.so.1", RTLD_LAZY | RTLD_LOCAL);
-    EGLint (*real_eglGetError)(void) = dlsym(egl, "eglGetError");
+    // void *egl = dlopen("libEGL.so.1", RTLD_LAZY | RTLD_LOCAL);
+    // EGLint (*real_eglGetError)(void) = dlsym(egl, "eglGetError");
 
-    if (extra_error != EGL_SUCCESS) {
-        EGLenum error = extra_error;
-        extra_error = EGL_SUCCESS;
-        return error;
-    }
+    // if (extra_error != EGL_SUCCESS) {
+    //     EGLenum error = extra_error;
+    //     extra_error = EGL_SUCCESS;
+    //     return error;
+    // }
 
-    assert(real_eglGetError);
-    return real_eglGetError();
+    // assert(real_eglGetError);
+    return eglGetError();
 }
 
 int
